{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.SeqUtils.IsoelectricPoint import IsoelectricPoint as IP\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "from io import StringIO\n",
    "import re\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15.768856699999986\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "# Search for a protein (e.g., \"rpIL\" in NCBI)\n",
    "Entrez.email = \"s230152@dtu.dk\"  # Required by NCBI\n",
    "\n",
    "# Search for the protein sequence\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"P0A7R1\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "# print(record)\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq\n",
    "\n",
    "# Print the raw record (it will contain the sequence)\n",
    "X=ProteinAnalysis(str(protein_sequence))\n",
    "print(X.molecular_weight()/1000)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IEP of peptide MQVILLDKVANLGSLGDQVNVKAGYARNFLVPQGKAVPATKKNIEFFEARRAELEAKLAEVLAAANARAEKINALETVTIASKAGDEGKLFGSIGTRDIADAVTAAGVEVAKSEVRLPNGVLRTTGEHEVSFQVHSEVFAKVIVNVVAE is 6.17\n",
      "It's charge at pH 7 is -1.29\n"
     ]
    }
   ],
   "source": [
    "protein = IP(protein_sequence) #rpil\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 7 is {:.2f}\".format(protein.charge_at_pH(7.0)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.esearch(db=\"protein\", term=\"uspA\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IEP of peptide MAYKHILVAVDLSPESEVLVSKAVSMAKPYNAKVSLIHVDVNYSDLYTGLIDVNLGDMQQRITEETSNSLKNLAKNSGYEIQEMLSGSGDLGQVLVDAIRKYDMDLVVCGHHQDFWSKLMSSARQLINTVHVDMLIVPLRDDDNA is 4.89\n",
      "It's charge at pH 7 is -0.72\n"
     ]
    }
   ],
   "source": [
    "protein = IP(protein_sequence)\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 7 is {:.2f}\".format(protein.charge_at_pH(5.0)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IEP of peptide TUFA is 5.18\n",
      "It's charge at pH 7 is -0.60\n"
     ]
    }
   ],
   "source": [
    "protein = IP(\"tufA\")\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 7 is {:.2f}\".format(protein.charge_at_pH(7.0)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       GFP_AEQVI                238 aa            linear   INV 02-OCT-2024\n",
      "DEFINITION  RecName: Full=Green fluorescent protein.\n",
      "ACCESSION   P42212\n",
      "VERSION     P42212.1\n",
      "DBSOURCE    UniProtKB: locus GFP_AEQVI, accession P42212;\n",
      "            class: standard.\n",
      "            extra accessions:Q17104,Q27903,Q93125\n",
      "            created: Nov 1, 1995.\n",
      "            sequence updated: Nov 1, 1995.\n",
      "            annotation updated: Oct 2, 2024.\n",
      "            xrefs: M62654.1, AAA27722.1, M62653.1, AAA27721.1, L29345.1,\n",
      "            AAA58246.1, X96418.1, CAA65278.1, U73901.1, AAB18957.1, JQ1514,\n",
      "            1B9C_A, 1B9C_B, 1B9C_C, 1B9C_D, 1BFP_A, 1C4F_A, 1CV7_A, 1EMA_A,\n",
      "            1EMB_A, 1EMC_A, 1EMC_B, 1EMC_C, 1EMC_D, 1EME_A, 1EMF_A, 1EMG_A,\n",
      "            1EMK_A, 1EML_A, 1EMM_A, 1F09_A, 1F0B_A, 1GFL_A, 1GFL_B, 1H6R_A,\n",
      "            1H6R_B, 1H6R_C, 1HCJ_A, 1HCJ_B, 1HCJ_C, 1HCJ_D, 1HUY_A, 1JBY_A,\n",
      "            1JBZ_A, 1JC0_A, 1JC0_B, 1JC0_C, 1JC1_A, 1JC1_B, 1JC1_C, 1KP5_A,\n",
      "            1KP5_B, 1KYP_A, 1KYR_A, 1KYS_A, 1MYW_A, 1Q4A_A, 1Q4B_A, 1Q4C_A,\n",
      "            1Q4D_A, 1Q4E_A, 1Q73_A, 1QXT_A, 1QY3_A, 1QYF_A, 1QYO_A, 1QYQ_A,\n",
      "            1RM9_A, 1RMM_A, 1RMO_A, 1RMP_A, 1RRX_A, 1S6Z_A, 1W7S_A, 1W7S_B,\n",
      "            1W7S_C, 1W7S_D, 1W7T_A, 1W7T_B, 1W7T_C, 1W7T_D, 1W7U_A, 1W7U_B,\n",
      "            1W7U_C, 1W7U_D, 1YFP_A, 1YFP_B, 1YHG_A, 1YHG_B, 1YHH_A, 1YHI_A,\n",
      "            1YJ2_A, 1YJF_A, 1Z1P_A, 1Z1Q_A, 2AH8_A, 2AH8_B, 2AHA_A, 2AHA_B,\n",
      "            2AWJ_A, 2AWK_A, 2AWL_A, 2AWM_A, 2B3P_A, 2B3Q_A, 2B3Q_B, 2B3Q_C,\n",
      "            2B3Q_D, 2DUE_A, 2DUF_A, 2DUG_A, 2DUH_A, 2DUI_A, 2EMD_A, 2EMN_A,\n",
      "            2EMO_A, 2FWQ_A, 2FZU_A, 2G16_A, 2G16_B, 2G2S_A, 2G2S_B, 2G3D_A,\n",
      "            2G3D_B, 2G5Z_A, 2G5Z_B, 2G6E_A, 2H6V_A, 2H9W_A, 2HCG_A, 2HFC_A,\n",
      "            2HGD_A, 2HGY_A, 2HJO_A, 2HQZ_A, 2HRS_A, 2JAD_A, 2O24_A, 2O29_A,\n",
      "            2O2B_A, 2OKW_A, 2OKW_B, 2OKW_C, 2OKW_D, 2OKW_E, 2OKW_F, 2OKY_A,\n",
      "            2OKY_B, 2Q57_A, 2Q6P_A, 2QRF_A, 2QT2_A, 2QU1_A, 2QZ0_A, 2WSN_A,\n",
      "            2WSO_A, 2WUR_A, 2Y0G_A, 2YDZ_A, 2YE0_A, 2YE1_A, 2YFP_A, 3AI4_A,\n",
      "            3AI5_A, 3CB9_A, 3CBE_A, 3CD1_A, 3CD9_A, 3DPW_A, 3DPX_A, 3DPZ_A,\n",
      "            3DQ1_A, 3DQ2_A, 3DQ3_A, 3DQ4_A, 3DQ5_A, 3DQ6_A, 3DQ7_A, 3DQ8_A,\n",
      "            3DQ9_A, 3DQA_A, 3DQC_A, 3DQD_A, 3DQE_A, 3DQF_A, 3DQH_A, 3DQI_A,\n",
      "            3DQJ_A, 3DQK_A, 3DQL_A, 3DQM_A, 3DQN_A, 3DQO_A, 3DQU_A, 3ED8_A,\n",
      "            3ED8_B, 3ED8_C, 3ED8_D, 3ED8_E, 3EK4_A, 3EK7_A, 3EK8_A, 3EKH_A,\n",
      "            3EKJ_A, 3EVP_A, 3EVR_A, 3EVU_A, 3EVV_A, 3G9A_A, 3GEX_A, 3GJ1_A,\n",
      "            3GJ1_B, 3GJ1_C, 3GJ1_D, 3GJ2_A, 3GJ2_B, 3GJ2_C, 3GJ2_D, 3I19_A,\n",
      "            3K1K_A, 3K1K_B, 3LA1_A, 3O77_A, 3O78_A, 3O78_B, 3OGO_A, 3OGO_B,\n",
      "            3OGO_C, 3OGO_D, 3OSQ_A, 3OSR_A, 3OSR_B, 3P28_A, 3SG2_A, 3SG3_A,\n",
      "            3SG4_A, 3SG5_A, 3SG6_A, 3SG7_A, 3SRY_A, 3SS0_A, 3SSH_A, 3SSK_A,\n",
      "            3SSL_A, 3SSP_A, 3SST_A, 3SSV_A, 3SSY_A, 3ST0_A, 3SV5_A, 3SVB_A,\n",
      "            3SVC_A, 3SVD_A, 3SVE_A, 3U8P_A, 3U8P_B, 3U8P_C, 3UFZ_A, 3UG0_A,\n",
      "            3V3D_A, 3VHT_A, 3W1C_A, 3W1D_A, 3WLC_A, 3WLD_A, 3ZTF_A, 4ANJ_A,\n",
      "            4AR7_A, 4AS8_A, 4B5Y_A, 4BDU_A, 4BDU_B, 4BDU_C, 4BDU_D, 4EN1_A,\n",
      "            4EN1_B, 4EUL_A, 4GES_B, 4GF6_B, 4H47_A, 4H48_A, 4IK1_A, 4IK3_A,\n",
      "            4IK4_A, 4IK5_A, 4IK8_A, 4IK9_A, 4J88_A, 4J88_B, 4J89_A, 4J89_B,\n",
      "            4J8A_A, 4JFG_A, 4JFG_B, 4JFG_C, 4JFG_D, 4JFG_E, 4JFG_F, 4JFG_G,\n",
      "            4JFG_H, 4JRB_A, 4KA9_A, 4KAG_A, 4KEX_A, 4KF5_A, 4KF5_B, 4KW4_A,\n",
      "            4KW8_A, 4KW9_A, 4L12_A, 4L13_A, 4L1I_A, 4LQT_A, 4LQU_A, 4LQU_B,\n",
      "            4LQU_C, 4LQU_D, 4LW5_A, 4LW5_B, 4LW5_C, 4LW5_D, 4LW5_E, 4N3D_A,\n",
      "            4N3D_B, 4NDJ_A, 4NDK_A, 4NDK_B, 4OGS_A, 4OGS_B, 4ORN_A, 4ORN_B,\n",
      "            4P1Q_A, 4P7H_A, 4P7H_B, 4PA0_A, 4PA0_B, 4PFE_A, 4PFE_B, 4U2V_A,\n",
      "            4U2V_B, 4U2V_C, 4U2V_D, 4XBI_A, 4XBI_B, 4XGY_A, 4XL5_A, 4XOV_A,\n",
      "            4XOW_A, 4XVP_A, 4XVP_B, 4XVP_C, 4Z4K_A, 4Z4K_B, 4Z4M_A, 4Z4M_B,\n",
      "            4ZF3_A, 4ZF3_B, 4ZF4_A, 4ZF4_B, 4ZF5_A, 4ZF5_B, 5AQB_B, 5B61_A,\n",
      "            5B61_B, 5B61_C, 5B61_D, 5B61_E, 5B61_F, 5BKF_E, 5BKG_E, 5DPG_A,\n",
      "            5DPH_A, 5DPH_B, 5DPI_A, 5DPI_B, 5DPI_C, 5DPI_D, 5DPI_E, 5DPI_F,\n",
      "            5DPJ_A, 5DPJ_B, 5DPJ_C, 5DPJ_D, 5DQB_A, 5DQM_A, 5DRF_A, 5DRG_A,\n",
      "            5DTX_A, 5DTY_A, 5DTZ_A, 5DTZ_B, 5DTZ_C, 5DTZ_D, 5DU0_A, 5DU0_B,\n",
      "            5DU0_C, 5DU0_D, 5EHU_A, 5EHU_B, 5F9G_A, 5FGU_A, 5HBD_A, 5HGE_A,\n",
      "            5HW9_A, 5HZO_A, 5HZO_B, 5J2O_A, 5J3N_A, 5J3N_B, 5JZK_A, 5JZK_B,\n",
      "            5JZL_A, 5JZL_B, 5KTG_A, 5KTG_B, 5LEL_C, 5LEL_F, 5LEL_I, 5LEM_C,\n",
      "            5MA3_B, 5MA4_B, 5MA5_B, 5MA5_D, 5MA6_A, 5MA8_B, 5MA8_D, 5MA9_B,\n",
      "            5MA9_D, 5MA9_F, 5MA9_H, 5MAD_B, 5MAD_D, 5MAD_F, 5MAD_H, 5MAK_B,\n",
      "            5MAK_D, 5MSE_A, 5MSE_B, 5MSE_C, 5MSE_D, 5N9O_A, 5NHN_A, 5NHN_B,\n",
      "            5NI3_A, 5NI3_B, 5NI3_C, 5NI3_D, 5O89_A, 5O8B_A, 5O8C_A, 5OX8_A,\n",
      "            5OX9_A, 5OXA_A, 5OXB_A, 5OXC_A, 5T3I_A, 5WJ2_A, 5WJ2_B, 5WJ3_A,\n",
      "            5WJ3_B, 5WJ4_A, 5WJ4_B, 5Z6Y_A, 6AA2_A, 6AA2_B, 6AA2_C, 6AA6_A,\n",
      "            6AA6_B, 6AS9_A, 6B7R_A, 6B7R_B, 6B7T_A, 6B7T_B, 6B9C_A, 6DQ0_A,\n",
      "            6DQ0_B, 6DQ1_A, 6EFR_A, 6FLL_A, 6FWW_A, 6GEL_A, 6GEL_B, 6GEZ_A,\n",
      "            6GEZ_B, 6GO8_A, 6GO9_A, 6GQG_A, 6GQH_A, 6GRM_A, 6HR1_A, 6HR1_B,\n",
      "            6HUT_A, 6IR6_A, 6IR7_A, 6ITC_G, 6JGH_A, 6JGI_A, 6JGJ_A, 6KKZ_A,\n",
      "            6KL0_A, 6KL1_A, 6KRG_A, 6L27_A, 6LNP_B, 6LNP_D, 6LR7_A, 6M63_A,\n",
      "            6M63_B, 6MB2_AA, 6MB2_BB, 6MB2_CC, 6MB2_DD, 6MB2_EE, 6MB2_FF,\n",
      "            6MB2_GG, 6MB2_HH, 6MB2_II, 6MB2_JJ, 6MB2_KK, 6MB2_LL, 6MB2_MM,\n",
      "            6MB2_NN, 6MB2_OO, 6MDR_BB, 6MDR_DD, 6MDR_FF, 6MDR_HH, 6MDR_JJ,\n",
      "            6MDR_LL, 6MDR_NN, 6MDR_PP, 6MKS_A, 6MKS_B, 6MKS_C, 6MKS_D, 6MKS_E,\n",
      "            6MKS_F, 6MKS_G, 6MKS_H, 6MKS_I, 6MKS_J, 6MKS_K, 6MKS_L, 6MKS_M,\n",
      "            6MKS_N, 6MKS_O, 6MKS_P, 6MKS_Q, 6MKS_R, 6MKS_S, 6MKS_T, 6MKS_U,\n",
      "            6MKS_V, 6MKS_W, 6MKS_X, 6MKS_Y, 6MKS_Z, 6MKS_AA, 6MKS_BB, 6MKS_CC,\n",
      "            6MKS_DD, 6MWQ_G, 6MWQ_H, 6MWQ_I, 6MWQ_J, 6OA8_A, 6OFK_A, 6OFK_B,\n",
      "            6OFL_A, 6OFL_B, 6OFM_A, 6OFM_B, 6OFN_A, 6OFN_B, 6OFO_A, 6OFO_B,\n",
      "            6OG8_A, 6OG8_B, 6OG9_A, 6OG9_B, 6OGA_A, 6OGA_B, 6OGB_A, 6OGB_B,\n",
      "            6OGC_A, 6OGC_B, 6PFR_A, 6PFS_A, 6PFT_A, 6PFU_A, 6QQ8_A, 6QQ9_A,\n",
      "            6QQA_A, 6QQB_A, 6QUH_A, 6QUH_C, 6QUI_A, 6QUI_D, 6QUJ_A, 6QUJ_D,\n",
      "            6SM0_A, 6T39_A, 6T3A_A, 6T90_K, 6UHJ_A, 6UHK_A, 6UHK_B, 6UHL_A,\n",
      "            6UHL_B, 6UHM_A, 6UHM_B, 6UHN_A, 6UHN_B, 6UHO_A, 6UHO_B, 6UHP_A,\n",
      "            6UHP_B, 6UHQ_A, 6UHR_A, 6UHR_B, 6UN4_A, 6UN5_A, 6UN5_B, 6UN6_A,\n",
      "            6UN6_B, 6UN7_A, 6UN7_B, 6UZ0_A, 6UZ3_A, 6VAL_A, 6VAL_B, 6VAL_C,\n",
      "            6VAL_D, 6VAL_E, 6VAL_F, 6VAL_G, 6VAL_H, 6VAL_I, 6VAL_J, 6VAL_K,\n",
      "            6VAM_A, 6VAM_B, 6VAM_C, 6VAM_D, 6VAM_E, 6VAM_F, 6VAM_G, 6VAM_H,\n",
      "            6VIO_A, 6VIO_B, 6VIO_C, 6VIO_D, 6VIO_E, 6VIO_F, 6VIO_G, 6VIO_H,\n",
      "            6WRF_S, 6WSG_S, 6WV3_A, 6WV4_A, 6WV5_A, 6WV6_A, 6WV7_A, 6WV7_B,\n",
      "            6WV8_A, 6WV9_A, 6WVA_A, 6WVB_A, 6WVD_A, 6WVE_A, 6WVF_A, 6WVG_A,\n",
      "            6WVG_B, 6WVH_A, 6WVH_B, 6WVI_A, 6XZF_A, 6YOV_K, 6ZUI_A, 7A7K_A,\n",
      "            7A7L_A, 7AA5_A, 7AA5_B, 7AA5_C, 7AA5_D, 7AMB_A, 7AMB_B, 7AMB_C,\n",
      "            7AMB_D, 7AMF_A, 7AMF_B, 7AMF_C, 7AMF_D, 7AMU_A, 7BWN_A, 7BWN_C,\n",
      "            7BWN_E, 7BWN_F, 7BWN_H, 7BWN_J, 7BWN_M, 7BWN_O, 7BYL_A, 7BYL_C,\n",
      "            7BYL_E, 7BYL_G, 7BYM_A, 7BYM_C, 7BYM_E, 7BYM_G, 7BYN_A, 7BYN_C,\n",
      "            7BYN_E, 7BYN_G, 7CD7_B, 7CD7_D, 7CD8_B, 7EDJ_I, 7EDJ_J, 7EDJ_K,\n",
      "            7FHK_A, 7FHK_C, 7FHL_A, 7FHL_C, 7FHN_A, 7FHN_C, 7FHO_A, 7FHO_C,\n",
      "            7K18_A, 7KM4_A, 7KS0_A, 7KS0_C, 7KS3_A, 7KS3_C, 7KUY_E, 7L31_E,\n",
      "            7O7C_A, 7O7D_A, 7O7E_A, 7O7H_A, 7O7U_A, 7O7V_A, 7O7W_A, 7O7X_A,\n",
      "            7PCA_A, 7PCZ_A, 7PCZ_B, 7PD0_A, 7PD0_B, 7PD0_C, 7PD0_D, 7PHR_D,\n",
      "            7PNN_A, 7SAH_A, 7SAI_A, 7SQY_A, 7SQY_B, 7SQY_C, 7SSV_A, 7SSV_B,\n",
      "            7SSV_C, 7SSV_D, 7SSX_A, 7SSX_B, 7SSX_C, 7SSX_D, 7SSY_A, 7SSY_B,\n",
      "            7SSY_C, 7SSY_D, 7SSZ_A, 7SSZ_B, 7SSZ_C, 7SSZ_D, 7SUN_A, 7SUN_B,\n",
      "            7X5V_A, 7X5V_B, 7X5V_C, 7X5V_D, 7YDQ_A, 7YEU_A, 7YV3_A, 7YV5_A,\n",
      "            8A6G_A, 8A6N_A, 8A6O_A, 8A6P_A, 8A6Q_A, 8A6R_A, 8A6S_A, 8A7V_A,\n",
      "            8A83_A, 8AHA_A, 8AHB_A, 8AM4_A, 8B6S_A, 8B6S_B, 8B6T_A, 8B6T_B,\n",
      "            8BAN_A, 8BAN_B, 8BAV_A, 8BAV_B, 8BVG_A, 8BVG_B, 8BVG_C, 8BVG_D,\n",
      "            8BVG_E, 8BVG_F, 8BXP_A, 8BXP_B, 8C1X_A, 8C1X_B, 8C1X_C, 8C1X_D,\n",
      "            8C7I_A, 8DFL_A, 8DFL_B, 8DFL_C, 8DFL_D, 8DHR_A, 8DHR_B, 8DN2_E,\n",
      "            8DN3_E, 8DN4_E, 8DN5_E, 8DPD_A, 8DTA_A, 8ET3_S, 8F6P_A, 8FCK_E,\n",
      "            8FED_G, 8FED_H, 8FEE_G, 8FEE_H, 8FEF_G, 8FEF_H, 8G0I_A, 8G4E_B,\n",
      "            8HCO_G, 8J0J_A, 8J0J_B, 8J0J_C, 8J1E_A, 8J1E_B, 8J1E_C, 8OTS_K,\n",
      "            8OVN_A, 8OVO_A, 8OVO_B, 8OVP_A, 8OVP_B, 8OVY_A, 8OVY_B, 8SFS_A,\n",
      "            8SFS_B, 8SFV_A, 8SFX_A, 8SFX_B, 8SFZ_D, 8SFZ_E, 8SFZ_F, 8SG3_A,\n",
      "            8SLC_A, 8SLC_B, 8SMU_A, 8SMU_B, 8SMU_C, 8SMU_D, 8SYG_A, 8SYG_B,\n",
      "            8SYG_C, 8SYG_D, 8SYG_E, 8SYG_F, 8SYG_G, 8SYG_H, 8SYG_I, 8SYG_J,\n",
      "            8SYG_K, 8SYG_L, 8SYG_M, 8SYG_N, 8T15_A, 8T15_B, 8T15_C, 8T15_D,\n",
      "            8T15_E, 8T15_F, 8T15_G, 8T15_H, 8T15_I, 8T15_J, 8T15_K, 8T15_L,\n",
      "            8T17_A, 8T17_B, 8T17_C, 8T17_D, 8T17_E, 8T17_F, 8T17_G, 8T17_H,\n",
      "            8T17_I, 8T17_J, 8T17_K, 8T17_L, 8T17_M, 8T17_N, 8T18_A, 8T18_B,\n",
      "            8T18_C, 8T18_D, 8T18_E, 8T18_F, 8T18_G, 8T18_H, 8T18_I, 8T18_J,\n",
      "            8T18_K, 8T18_L, 8T6K_A, 8T6K_B, 8T6K_C, 8T6K_D, 8T6K_E, 8T6K_F,\n",
      "            8T6K_G, 8T6K_H, 8T6K_I, 8T6K_J, 8T6K_K, 8T6K_L, 8T6K_M, 8T6K_N,\n",
      "            8T6L_A, 8T6Q_A, 8T6Q_B, 8T6Q_C, 8T6Q_D, 8T6Q_F, 8T6Q_G, 8T6Q_H,\n",
      "            8T6Q_I, 8T6Q_J, 8T6Q_K, 8T6Q_L, 8T6Q_N, 8TLM_C, 8TU9_A, 8TU9_B,\n",
      "            8U4N_R, 8U4O_R, 8U4P_R, 8U4Q_R, 8U4R_R, 8U4S_C, 8U4S_G, 8U4S_R,\n",
      "            8W2L_A, 8W2L_B, 8W2L_C, 8W2L_D\n",
      "            xrefs (non-sequence databases): PDBsum:1B9C, PDBsum:1BFP,\n",
      "            PDBsum:1C4F, PDBsum:1CV7, PDBsum:1EMA, PDBsum:1EMB, PDBsum:1EMC,\n",
      "            PDBsum:1EME, PDBsum:1EMF, PDBsum:1EMG, PDBsum:1EMK, PDBsum:1EML,\n",
      "            PDBsum:1EMM, PDBsum:1F09, PDBsum:1F0B, PDBsum:1GFL, PDBsum:1H6R,\n",
      "            PDBsum:1HCJ, PDBsum:1HUY, PDBsum:1JBY, PDBsum:1JBZ, PDBsum:1JC0,\n",
      "            PDBsum:1JC1, PDBsum:1KP5, PDBsum:1KYP, PDBsum:1KYR, PDBsum:1KYS,\n",
      "            PDBsum:1MYW, PDBsum:1Q4A, PDBsum:1Q4B, PDBsum:1Q4C, PDBsum:1Q4D,\n",
      "            PDBsum:1Q4E, PDBsum:1Q73, PDBsum:1QXT, PDBsum:1QY3, PDBsum:1QYF,\n",
      "            PDBsum:1QYO, PDBsum:1QYQ, PDBsum:1RM9, PDBsum:1RMM, PDBsum:1RMO,\n",
      "            PDBsum:1RMP, PDBsum:1RRX, PDBsum:1S6Z, PDBsum:1W7S, PDBsum:1W7T,\n",
      "            PDBsum:1W7U, PDBsum:1YFP, PDBsum:1YHG, PDBsum:1YHH, PDBsum:1YHI,\n",
      "            PDBsum:1YJ2, PDBsum:1YJF, PDBsum:1Z1P, PDBsum:1Z1Q, PDBsum:2AH8,\n",
      "            PDBsum:2AHA, PDBsum:2AWJ, PDBsum:2AWK, PDBsum:2AWL, PDBsum:2AWM,\n",
      "            PDBsum:2B3P, PDBsum:2B3Q, PDBsum:2DUE, PDBsum:2DUF, PDBsum:2DUG,\n",
      "            PDBsum:2DUH, PDBsum:2DUI, PDBsum:2EMD, PDBsum:2EMN, PDBsum:2EMO,\n",
      "            PDBsum:2FWQ, PDBsum:2FZU, PDBsum:2G16, PDBsum:2G2S, PDBsum:2G3D,\n",
      "            PDBsum:2G5Z, PDBsum:2G6E, PDBsum:2H6V, PDBsum:2H9W, PDBsum:2HCG,\n",
      "            PDBsum:2HFC, PDBsum:2HGD, PDBsum:2HGY, PDBsum:2HJO, PDBsum:2HQZ,\n",
      "            PDBsum:2HRS, PDBsum:2JAD, PDBsum:2O24, PDBsum:2O29, PDBsum:2O2B,\n",
      "            PDBsum:2OKW, PDBsum:2OKY, PDBsum:2Q57, PDBsum:2Q6P, PDBsum:2QRF,\n",
      "            PDBsum:2QT2, PDBsum:2QU1, PDBsum:2QZ0, PDBsum:2WSN, PDBsum:2WSO,\n",
      "            PDBsum:2WUR, PDBsum:2Y0G, PDBsum:2YDZ, PDBsum:2YE0, PDBsum:2YE1,\n",
      "            PDBsum:2YFP, PDBsum:3AI4, PDBsum:3AI5, PDBsum:3CB9, PDBsum:3CBE,\n",
      "            PDBsum:3CD1, PDBsum:3CD9, PDBsum:3DPW, PDBsum:3DPX, PDBsum:3DPZ,\n",
      "            PDBsum:3DQ1, PDBsum:3DQ2, PDBsum:3DQ3, PDBsum:3DQ4, PDBsum:3DQ5,\n",
      "            PDBsum:3DQ6, PDBsum:3DQ7, PDBsum:3DQ8, PDBsum:3DQ9, PDBsum:3DQA,\n",
      "            PDBsum:3DQC, PDBsum:3DQD, PDBsum:3DQE, PDBsum:3DQF, PDBsum:3DQH,\n",
      "            PDBsum:3DQI, PDBsum:3DQJ, PDBsum:3DQK, PDBsum:3DQL, PDBsum:3DQM,\n",
      "            PDBsum:3DQN, PDBsum:3DQO, PDBsum:3DQU, PDBsum:3ED8, PDBsum:3EK4,\n",
      "            PDBsum:3EK7, PDBsum:3EK8, PDBsum:3EKH, PDBsum:3EKJ, PDBsum:3EVP,\n",
      "            PDBsum:3EVR, PDBsum:3EVU, PDBsum:3EVV, PDBsum:3G9A, PDBsum:3GEX,\n",
      "            PDBsum:3GJ1, PDBsum:3GJ2, PDBsum:3I19, PDBsum:3K1K, PDBsum:3LA1,\n",
      "            PDBsum:3O77, PDBsum:3O78, PDBsum:3OGO, PDBsum:3OSQ, PDBsum:3OSR,\n",
      "            PDBsum:3P28, PDBsum:3SG2, PDBsum:3SG3, PDBsum:3SG4, PDBsum:3SG5,\n",
      "            PDBsum:3SG6, PDBsum:3SG7, PDBsum:3SRY, PDBsum:3SS0, PDBsum:3SSH,\n",
      "            PDBsum:3SSK, PDBsum:3SSL, PDBsum:3SSP, PDBsum:3SST, PDBsum:3SSV,\n",
      "            PDBsum:3SSY, PDBsum:3ST0, PDBsum:3SV5, PDBsum:3SVB, PDBsum:3SVC,\n",
      "            PDBsum:3SVD, PDBsum:3SVE, PDBsum:3U8P, PDBsum:3UFZ, PDBsum:3UG0,\n",
      "            PDBsum:3V3D, PDBsum:3VHT, PDBsum:3W1C, PDBsum:3W1D, PDBsum:3WLC,\n",
      "            PDBsum:3WLD, PDBsum:3ZTF, PDBsum:4ANJ, PDBsum:4AR7, PDBsum:4AS8,\n",
      "            PDBsum:4B5Y, PDBsum:4BDU, PDBsum:4EN1, PDBsum:4EUL, PDBsum:4GES,\n",
      "            PDBsum:4GF6, PDBsum:4H47, PDBsum:4H48, PDBsum:4IK1, PDBsum:4IK3,\n",
      "            PDBsum:4IK4, PDBsum:4IK5, PDBsum:4IK8, PDBsum:4IK9, PDBsum:4J88,\n",
      "            PDBsum:4J89, PDBsum:4J8A, PDBsum:4JFG, PDBsum:4JRB, PDBsum:4KA9,\n",
      "            PDBsum:4KAG, PDBsum:4KEX, PDBsum:4KF5, PDBsum:4KW4, PDBsum:4KW8,\n",
      "            PDBsum:4KW9, PDBsum:4L12, PDBsum:4L13, PDBsum:4L1I, PDBsum:4LQT,\n",
      "            PDBsum:4LQU, PDBsum:4LW5, PDBsum:4N3D, PDBsum:4NDJ, PDBsum:4NDK,\n",
      "            PDBsum:4OGS, PDBsum:4ORN, PDBsum:4P1Q, PDBsum:4P7H, PDBsum:4PA0,\n",
      "            PDBsum:4PFE, PDBsum:4U2V, PDBsum:4XBI, PDBsum:4XGY, PDBsum:4XL5,\n",
      "            PDBsum:4XOV, PDBsum:4XOW, PDBsum:4XVP, PDBsum:4Z4K, PDBsum:4Z4M,\n",
      "            PDBsum:4ZF3, PDBsum:4ZF4, PDBsum:4ZF5, PDBsum:5AQB, PDBsum:5B61,\n",
      "            PDBsum:5BKF, PDBsum:5BKG, PDBsum:5DPG, PDBsum:5DPH, PDBsum:5DPI,\n",
      "            PDBsum:5DPJ, PDBsum:5DQB, PDBsum:5DQM, PDBsum:5DRF, PDBsum:5DRG,\n",
      "            PDBsum:5DTX, PDBsum:5DTY, PDBsum:5DTZ, PDBsum:5DU0, PDBsum:5EHU,\n",
      "            PDBsum:5F9G, PDBsum:5FGU, PDBsum:5HBD, PDBsum:5HGE, PDBsum:5HW9,\n",
      "            PDBsum:5HZO, PDBsum:5J2O, PDBsum:5J3N, PDBsum:5JZK, PDBsum:5JZL,\n",
      "            PDBsum:5KTG, PDBsum:5LEL, PDBsum:5LEM, PDBsum:5MA3, PDBsum:5MA4,\n",
      "            PDBsum:5MA5, PDBsum:5MA6, PDBsum:5MA8, PDBsum:5MA9, PDBsum:5MAD,\n",
      "            PDBsum:5MAK, PDBsum:5MSE, PDBsum:5N9O, PDBsum:5NHN, PDBsum:5NI3,\n",
      "            PDBsum:5O89, PDBsum:5O8B, PDBsum:5O8C, PDBsum:5OX8, PDBsum:5OX9,\n",
      "            PDBsum:5OXA, PDBsum:5OXB, PDBsum:5OXC, PDBsum:5T3I, PDBsum:5WJ2,\n",
      "            PDBsum:5WJ3, PDBsum:5WJ4, PDBsum:5Z6Y, PDBsum:6AA2, PDBsum:6AA6,\n",
      "            PDBsum:6AS9, PDBsum:6B7R, PDBsum:6B7T, PDBsum:6B9C, PDBsum:6DQ0,\n",
      "            PDBsum:6DQ1, PDBsum:6EFR, PDBsum:6FLL, PDBsum:6FWW, PDBsum:6GEL,\n",
      "            PDBsum:6GEZ, PDBsum:6GO8, PDBsum:6GO9, PDBsum:6GQG, PDBsum:6GQH,\n",
      "            PDBsum:6GRM, PDBsum:6HR1, PDBsum:6HUT, PDBsum:6IR6, PDBsum:6IR7,\n",
      "            PDBsum:6ITC, PDBsum:6JGH, PDBsum:6JGI, PDBsum:6JGJ, PDBsum:6KKZ,\n",
      "            PDBsum:6KL0, PDBsum:6KL1, PDBsum:6KRG, PDBsum:6L27, PDBsum:6LNP,\n",
      "            PDBsum:6LR7, PDBsum:6M63, PDBsum:6MB2, PDBsum:6MDR, PDBsum:6MKS,\n",
      "            PDBsum:6MWQ, PDBsum:6OA8, PDBsum:6OFK, PDBsum:6OFL, PDBsum:6OFM,\n",
      "            PDBsum:6OFN, PDBsum:6OFO, PDBsum:6OG8, PDBsum:6OG9, PDBsum:6OGA,\n",
      "            PDBsum:6OGB, PDBsum:6OGC, PDBsum:6PFR, PDBsum:6PFS, PDBsum:6PFT,\n",
      "            PDBsum:6PFU, PDBsum:6QQ8, PDBsum:6QQ9, PDBsum:6QQA, PDBsum:6QQB,\n",
      "            PDBsum:6QUH, PDBsum:6QUI, PDBsum:6QUJ, PDBsum:6SM0, PDBsum:6T39,\n",
      "            PDBsum:6T3A, PDBsum:6T90, PDBsum:6UHJ, PDBsum:6UHK, PDBsum:6UHL,\n",
      "            PDBsum:6UHM, PDBsum:6UHN, PDBsum:6UHO, PDBsum:6UHP, PDBsum:6UHQ,\n",
      "            PDBsum:6UHR, PDBsum:6UN4, PDBsum:6UN5, PDBsum:6UN6, PDBsum:6UN7,\n",
      "            PDBsum:6UZ0, PDBsum:6UZ3, PDBsum:6VAL, PDBsum:6VAM, PDBsum:6VIO,\n",
      "            PDBsum:6WRF, PDBsum:6WSG, PDBsum:6WV3, PDBsum:6WV4, PDBsum:6WV5,\n",
      "            PDBsum:6WV6, PDBsum:6WV7, PDBsum:6WV8, PDBsum:6WV9, PDBsum:6WVA,\n",
      "            PDBsum:6WVB, PDBsum:6WVD, PDBsum:6WVE, PDBsum:6WVF, PDBsum:6WVG,\n",
      "            PDBsum:6WVH, PDBsum:6WVI, PDBsum:6XZF, PDBsum:6YOV, PDBsum:6ZUI,\n",
      "            PDBsum:7A7K, PDBsum:7A7L, PDBsum:7AA5, PDBsum:7AMB, PDBsum:7AMF,\n",
      "            PDBsum:7AMU, PDBsum:7BWN, PDBsum:7BYL, PDBsum:7BYM, PDBsum:7BYN,\n",
      "            PDBsum:7CD7, PDBsum:7CD8, PDBsum:7EDJ, PDBsum:7FHK, PDBsum:7FHL,\n",
      "            PDBsum:7FHN, PDBsum:7FHO, PDBsum:7K18, PDBsum:7KM4, PDBsum:7KS0,\n",
      "            PDBsum:7KS3, PDBsum:7KUY, PDBsum:7L31, PDBsum:7O7C, PDBsum:7O7D,\n",
      "            PDBsum:7O7E, PDBsum:7O7H, PDBsum:7O7U, PDBsum:7O7V, PDBsum:7O7W,\n",
      "            PDBsum:7O7X, PDBsum:7PCA, PDBsum:7PCZ, PDBsum:7PD0, PDBsum:7PHR,\n",
      "            PDBsum:7PNN, PDBsum:7SAH, PDBsum:7SAI, PDBsum:7SQY, PDBsum:7SSV,\n",
      "            PDBsum:7SSX, PDBsum:7SSY, PDBsum:7SSZ, PDBsum:7SUN, PDBsum:7X5V,\n",
      "            PDBsum:7YDQ, PDBsum:7YEU, PDBsum:7YV3, PDBsum:7YV5, PDBsum:8A6G,\n",
      "            PDBsum:8A6N, PDBsum:8A6O, PDBsum:8A6P, PDBsum:8A6Q, PDBsum:8A6R,\n",
      "            PDBsum:8A6S, PDBsum:8A7V, PDBsum:8A83, PDBsum:8AHA, PDBsum:8AHB,\n",
      "            PDBsum:8AM4, PDBsum:8B6S, PDBsum:8B6T, PDBsum:8BAN, PDBsum:8BAV,\n",
      "            PDBsum:8BVG, PDBsum:8BXP, PDBsum:8C1X, PDBsum:8C7I, PDBsum:8DFL,\n",
      "            PDBsum:8DHR, PDBsum:8DN2, PDBsum:8DN3, PDBsum:8DN4, PDBsum:8DN5,\n",
      "            PDBsum:8DPD, PDBsum:8DTA, PDBsum:8ET3, PDBsum:8F6P, PDBsum:8FCK,\n",
      "            PDBsum:8FED, PDBsum:8FEE, PDBsum:8FEF, PDBsum:8G0I, PDBsum:8G4E,\n",
      "            PDBsum:8HCO, PDBsum:8J0J, PDBsum:8J1E, PDBsum:8OTS, PDBsum:8OVN,\n",
      "            PDBsum:8OVO, PDBsum:8OVP, PDBsum:8OVY, PDBsum:8SFS, PDBsum:8SFV,\n",
      "            PDBsum:8SFX, PDBsum:8SFZ, PDBsum:8SG3, PDBsum:8SLC, PDBsum:8SMU,\n",
      "            PDBsum:8SYG, PDBsum:8T15, PDBsum:8T17, PDBsum:8T18, PDBsum:8T6K,\n",
      "            PDBsum:8T6L, PDBsum:8T6Q, PDBsum:8TLM, PDBsum:8TU9, PDBsum:8U4N,\n",
      "            PDBsum:8U4O, PDBsum:8U4P, PDBsum:8U4Q, PDBsum:8U4R, PDBsum:8U4S,\n",
      "            PDBsum:8W2L, AlphaFoldDB:P42212, BMRB:P42212, EMDB:EMD-10406,\n",
      "            EMDB:EMD-10864, EMDB:EMD-11690, EMDB:EMD-13427, EMDB:EMD-20949,\n",
      "            EMDB:EMD-20951, EMDB:EMD-21142, EMDB:EMD-21143, EMDB:EMD-21882,\n",
      "            EMDB:EMD-21892, EMDB:EMD-22621, EMDB:EMD-23014, EMDB:EMD-23015,\n",
      "            EMDB:EMD-23041, EMDB:EMD-23148, EMDB:EMD-25414, EMDB:EMD-25416,\n",
      "            EMDB:EMD-25417, EMDB:EMD-28585, EMDB:EMD-28887, EMDB:EMD-28981,\n",
      "            EMDB:EMD-29023, EMDB:EMD-29024, EMDB:EMD-29025, EMDB:EMD-29718,\n",
      "            EMDB:EMD-33756, EMDB:EMD-35904, EMDB:EMD-35920, EMDB:EMD-36753,\n",
      "            EMDB:EMD-41071, EMDB:EMD-41370, EMDB:EMD-9064, EMDB:EMD-9104,\n",
      "            EMDB:EMD-9277, EMDB:EMD-9374, EMDB:EMD-9403, EMDB:EMD-9404,\n",
      "            EMDB:EMD-9731, SMR:P42212, IntAct:P42212, ABCD:P42212,\n",
      "            BioCyc:MetaCyc:MONOMER-20291, EvolutionaryTrace:P42212, GO:0008218,\n",
      "            GO:0006091, Gene3D:2.40.155.10, InterPro:IPR009017,\n",
      "            InterPro:IPR011584, InterPro:IPR000786, Pfam:PF01353,\n",
      "            PRINTS:PR01229, SUPFAM:SSF54511\n",
      "KEYWORDS    3D-structure; Chromophore; Direct protein sequencing; Luminescence;\n",
      "            Photoprotein.\n",
      "SOURCE      Aequorea victoria\n",
      "  ORGANISM  Aequorea victoria\n",
      "            Eukaryota; Metazoa; Cnidaria; Hydrozoa; Hydroidolina; Leptothecata;\n",
      "            Aequoreidae; Aequorea.\n",
      "REFERENCE   1  (residues 1 to 238)\n",
      "  AUTHORS   Prasher,D.C., Eckenrode,V.K., Ward,W.W., Prendergast,F.G. and\n",
      "            Cormier,M.J.\n",
      "  TITLE     Primary structure of the Aequorea victoria green-fluorescent\n",
      "            protein\n",
      "  JOURNAL   Gene 111 (2), 229-233 (1992)\n",
      "   PUBMED   1347277\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA], PARTIAL PROTEIN SEQUENCE, AND VARIANTS\n",
      "            TYR-100; SER-108; MET-141 AND ILE-219.\n",
      "REFERENCE   2  (residues 1 to 238)\n",
      "  AUTHORS   Inouye,S. and Tsuji,F.I.\n",
      "  TITLE     Aequorea green fluorescent protein. Expression of the gene and\n",
      "            fluorescence characteristics of the recombinant protein\n",
      "  JOURNAL   FEBS Lett 341 (2-3), 277-280 (1994)\n",
      "   PUBMED   8137953\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA], AND BIOPHYSICOCHEMICAL PROPERTIES.\n",
      "REFERENCE   3  (residues 1 to 238)\n",
      "  AUTHORS   Rouwendal,G.J., Mendes,O., Wolbert,E.J. and Douwe de Boer,A.\n",
      "  TITLE     Enhanced expression in tobacco of the gene encoding green\n",
      "            fluorescent protein by modification of its codon usage\n",
      "  JOURNAL   Plant Mol Biol 33 (6), 989-999 (1997)\n",
      "   PUBMED   9154981\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA], BIOPHYSICOCHEMICAL PROPERTIES, AND\n",
      "            BIOTECHNOLOGY.\n",
      "REFERENCE   4  (residues 1 to 238)\n",
      "  AUTHORS   Cormack,B.P., Valdivia,R.H. and Falkow,S.\n",
      "  TITLE     FACS-optimized mutants of the green fluorescent protein (GFP)\n",
      "  JOURNAL   Gene 173 (1 Spec No), 33-38 (1996)\n",
      "   PUBMED   8707053\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA], BIOTECHNOLOGY, AND MUTAGENESIS\n",
      "            OF PHE-64; SER-65; VAL-68 AND SER-72.\n",
      "REFERENCE   5  (residues 1 to 238)\n",
      "  AUTHORS   Cody,C.W., Prasher,D.C., Westler,W.M., Prendergast,F.G. and\n",
      "            Ward,W.W.\n",
      "  TITLE     Chemical structure of the hexapeptide chromophore of the Aequorea\n",
      "            green-fluorescent protein\n",
      "  JOURNAL   Biochemistry 32 (5), 1212-1218 (1993)\n",
      "   PUBMED   8448132\n",
      "  REMARK    CHROMOPHORE, AND DEHYDROGENATION AT TYR-66.\n",
      "REFERENCE   6  (residues 1 to 238)\n",
      "  AUTHORS   Heim,R., Prasher,D.C. and Tsien,R.Y.\n",
      "  TITLE     Wavelength mutations and posttranslational autoxidation of green\n",
      "            fluorescent protein\n",
      "  JOURNAL   Proc Natl Acad Sci U S A 91 (26), 12501-12504 (1994)\n",
      "   PUBMED   7809066\n",
      "  REMARK    MUTAGENESIS OF TYR-66; ILE-167; SER-202 AND THR-203.\n",
      "REFERENCE   7  (residues 1 to 238)\n",
      "  AUTHORS   Delagrave,S., Hawtin,R.E., Silva,C.M., Yang,M.M. and Youvan,D.C.\n",
      "  TITLE     Red-shifted excitation mutants of the green fluorescent protein\n",
      "  JOURNAL   Biotechnology (N Y) 13 (2), 151-154 (1995)\n",
      "   PUBMED   9634755\n",
      "  REMARK    MUTAGENESIS OF PHE-64; SER-65 AND GLN-69.\n",
      "REFERENCE   8  (residues 1 to 238)\n",
      "  AUTHORS   Ehrig,T., O'Kane,D.J. and Prendergast,F.G.\n",
      "  TITLE     Green-fluorescent protein mutants with altered fluorescence\n",
      "            excitation spectra\n",
      "  JOURNAL   FEBS Lett 367 (2), 163-166 (1995)\n",
      "   PUBMED   7796912\n",
      "  REMARK    MUTAGENESIS OF THR-203 AND GLU-222.\n",
      "REFERENCE   9  (residues 1 to 238)\n",
      "  AUTHORS   Cubitt,A.B., Heim,R., Adams,S.R., Boyd,A.E., Gross,L.A. and\n",
      "            Tsien,R.Y.\n",
      "  TITLE     Understanding, improving and using green fluorescent proteins\n",
      "  JOURNAL   Trends Biochem Sci 20 (11), 448-455 (1995)\n",
      "   PUBMED   8578587\n",
      "  REMARK    MUTAGENESIS OF TYR-66, AND BIOTECHNOLOGY.\n",
      "REFERENCE   10 (residues 1 to 238)\n",
      "  AUTHORS   Siemering,K.R., Golbik,R., Sever,R. and Haseloff,J.\n",
      "  TITLE     Mutations that suppress the thermosensitivity of green fluorescent\n",
      "            protein\n",
      "  JOURNAL   Curr Biol 6 (12), 1653-1663 (1996)\n",
      "   PUBMED   8994830\n",
      "  REMARK    MUTAGENESIS OF VAL-163 AND SER-175.\n",
      "REFERENCE   11 (residues 1 to 238)\n",
      "  AUTHORS   Crameri,A., Whitehorn,E.A., Tate,E. and Stemmer,W.P.\n",
      "  TITLE     Improved green fluorescent protein by molecular evolution using DNA\n",
      "            shuffling\n",
      "  JOURNAL   Nat Biotechnol 14 (3), 315-319 (1996)\n",
      "   PUBMED   9630892\n",
      "  REMARK    MUTAGENESIS OF PHE-99; MET-153 AND VAL-163.\n",
      "REFERENCE   12 (residues 1 to 238)\n",
      "  AUTHORS   Yokoe,H. and Meyer,T.\n",
      "  TITLE     Spatial dynamics of GFP-tagged proteins investigated by local\n",
      "            fluorescence enhancement\n",
      "  JOURNAL   Nat Biotechnol 14 (10), 1252-1256 (1996)\n",
      "   PUBMED   9631088\n",
      "  REMARK    MUTAGENESIS OF PHE-99; MET-153 AND VAL-163.\n",
      "REFERENCE   13 (residues 1 to 238)\n",
      "  AUTHORS   Kimata,Y., Iwaki,M., Lim,C.R. and Kohno,K.\n",
      "  TITLE     A novel mutation which enhances the fluorescence of green\n",
      "            fluorescent protein at high temperatures\n",
      "  JOURNAL   Biochem Biophys Res Commun 232 (1), 69-73 (1997)\n",
      "   PUBMED   9125154\n",
      "  REMARK    MUTAGENESIS OF SER-147.\n",
      "REFERENCE   14 (residues 1 to 238)\n",
      "  AUTHORS   Patterson,G.H., Knobel,S.M., Sharif,W.D., Kain,S.R. and Piston,D.W.\n",
      "  TITLE     Use of the green fluorescent protein and its mutants in\n",
      "            quantitative fluorescence microscopy\n",
      "  JOURNAL   Biophys J 73 (5), 2782-2790 (1997)\n",
      "   PUBMED   9370472\n",
      "  REMARK    MUTAGENESIS OF PHE-64; SER-65; TYR-66; PHE-99; MET-153 AND VAL-163.\n",
      "REFERENCE   15 (residues 1 to 238)\n",
      "  AUTHORS   Tsien,R.Y.\n",
      "  TITLE     The green fluorescent protein\n",
      "  JOURNAL   Annu Rev Biochem 67, 509-544 (1998)\n",
      "   PUBMED   9759496\n",
      "  REMARK    REVIEW ON MUTAGENESIS, AND BIOTECHNOLOGY.\n",
      "REFERENCE   16 (residues 1 to 238)\n",
      "  AUTHORS   Srinivas,S., Watanabe,T., Lin,C.S., William,C.M., Tanabe,Y.,\n",
      "            Jessell,T.M. and Costantini,F.\n",
      "  TITLE     Cre reporter strains produced by targeted insertion of EYFP and\n",
      "            ECFP into the ROSA26 locus\n",
      "  JOURNAL   BMC Dev Biol 1, 4 (2001)\n",
      "   PUBMED   11299042\n",
      "  REMARK    MUTAGENESIS OF SER-65; VAL-68; SER-72; ASN-146; MET-153; VAL-163\n",
      "            AND THR-203.\n",
      "REFERENCE   17 (residues 1 to 238)\n",
      "  AUTHORS   Baumstark-Khan,C., Hellweg,C.E., Palm,M. and Horneck,G.\n",
      "  TITLE     Enhanced green fluorescent protein (EGFP) for space radiation\n",
      "            research using mammalian cells in the International Space Station\n",
      "  JOURNAL   Phys Med 17 Suppl 1, 210-214 (2001)\n",
      "   PUBMED   11776264\n",
      "  REMARK    MUTAGENESIS OF PHE-64 AND SER-65.\n",
      "REFERENCE   18 (residues 1 to 238)\n",
      "  AUTHORS   Zapata-Hommer,O. and Griesbeck,O.\n",
      "  TITLE     Efficiently folding and circularly permuted variants of the\n",
      "            Sapphire mutant of GFP\n",
      "  JOURNAL   BMC Biotechnol 3, 5 (2003)\n",
      "   PUBMED   12769828\n",
      "  REMARK    MUTAGENESIS OF SER-72; TYR-145 AND THR-203.\n",
      "REFERENCE   19 (residues 1 to 238)\n",
      "  AUTHORS   Wang,Z., Shah,J.V., Chen,Z., Sun,C.H. and Berns,M.W.\n",
      "  TITLE     Fluorescence correlation spectroscopy investigation of a GFP\n",
      "            mutant-enhanced cyan fluorescent protein and its tubulin fusion in\n",
      "            living cells with two-photon excitation\n",
      "  JOURNAL   J Biomed Opt 9 (2), 395-403 (2004)\n",
      "   PUBMED   15065907\n",
      "  REMARK    MUTAGENESIS OF PHE-64; SER-65; TYR-66; ASN-146; MET-153 AND\n",
      "            VAL-163.\n",
      "REFERENCE   20 (residues 1 to 238)\n",
      "  AUTHORS   Rizzo,M.A., Springer,G.H., Granada,B. and Piston,D.W.\n",
      "  TITLE     An improved cyan fluorescent protein variant useful for FRET\n",
      "  JOURNAL   Nat Biotechnol 22 (4), 445-449 (2004)\n",
      "   PUBMED   14990965\n",
      "  REMARK    MUTAGENESIS OF PHE-64; SER-65; TYR-66; SER-72; TYR-145; ASN-146;\n",
      "            HIS-148; MET-153 AND VAL-163.\n",
      "REFERENCE   21 (residues 1 to 238)\n",
      "  AUTHORS   Scruggs,A.W., Flores,C.L., Wachter,R. and Woodbury,N.W.\n",
      "  TITLE     Development and characterization of green fluorescent protein\n",
      "            mutants with altered lifetimes\n",
      "  JOURNAL   Biochemistry 44 (40), 13377-13384 (2005)\n",
      "   PUBMED   16201762\n",
      "  REMARK    MUTAGENESIS OF SER-30; PHE-46; GLN-69; ASP-103; TYR-145; VAL-176;\n",
      "            VAL-193 AND ASN-198.\n",
      "REFERENCE   22 (residues 1 to 238)\n",
      "  AUTHORS   Grailhe,R., Merola,F., Ridard,J., Couvignou,S., Le Poupon,C.,\n",
      "            Changeux,J.P. and Laguitton-Pasquier,H.\n",
      "  TITLE     Monitoring protein interactions in the living cell through the\n",
      "            fluorescence decays of the cyan fluorescent protein\n",
      "  JOURNAL   Chemphyschem 7 (7), 1442-1454 (2006)\n",
      "   PUBMED   16739159\n",
      "  REMARK    MUTAGENESIS OF ALA-206.\n",
      "REFERENCE   23 (residues 1 to 238)\n",
      "  AUTHORS   Teerawanichpan,P., Hoffman,T., Ashe,P., Datla,R. and Selvaraj,G.\n",
      "  TITLE     Investigations of combinations of mutations in the jellyfish green\n",
      "            fluorescent protein (GFP) that afford brighter fluorescence, and\n",
      "            use of a version (VisGreen) in plant, bacterial, and animal cells\n",
      "  JOURNAL   Biochim Biophys Acta 1770 (9), 1360-1368 (2007)\n",
      "   PUBMED   17658219\n",
      "  REMARK    MUTAGENESIS OF PHE-64; SER-65; SER-72; ASN-149; MET-153 AND\n",
      "            ILE-167.\n",
      "REFERENCE   24 (residues 1 to 238)\n",
      "  AUTHORS   Wong,F.H., Banks,D.S., Abu-Arish,A. and Fradin,C.\n",
      "  TITLE     A molecular thermometer based on fluorescent protein blinking\n",
      "  JOURNAL   J Am Chem Soc 129 (34), 10302-10303 (2007)\n",
      "   PUBMED   17685514\n",
      "  REMARK    BIOTECHNOLOGY.\n",
      "REFERENCE   25 (residues 1 to 238)\n",
      "  AUTHORS   Mishin,A.S., Subach,F.V., Yampolsky,I.V., King,W., Lukyanov,K.A.\n",
      "            and Verkhusha,V.V.\n",
      "  TITLE     The first mutant of the Aequorea victoria green fluorescent protein\n",
      "            that forms a red chromophore\n",
      "  JOURNAL   Biochemistry 47 (16), 4666-4673 (2008)\n",
      "   PUBMED   18366185\n",
      "  REMARK    MUTAGENESIS OF PHE-46; PHE-64; VAL-68; LYS-162; VAL-163; ILE-167\n",
      "            AND ILE-171.\n",
      "REFERENCE   26 (residues 1 to 238)\n",
      "  AUTHORS   Park,J.W. and Rhee,Y.M.\n",
      "  TITLE     Interpolated mechanics-molecular mechanics study of internal\n",
      "            rotation dynamics of the chromophore unit in blue fluorescent\n",
      "            protein and its variants\n",
      "  JOURNAL   J Phys Chem B 116 (36), 11137-11147 (2012)\n",
      "   PUBMED   22891786\n",
      "  REMARK    MUTAGENESIS OF SER-30; TYR-39; TYR-66; SER-72; GLN-80; ASN-105;\n",
      "            ILE-128; TYR-145; VAL-150; ASP-155; ILE-171; ASN-198; ALA-206 AND\n",
      "            VAL-224.\n",
      "REFERENCE   27 (residues 1 to 238)\n",
      "  AUTHORS   Yang,F., Moss,L.G. and Phillips,G.N. Jr.\n",
      "  TITLE     The molecular structure of green fluorescent protein\n",
      "  JOURNAL   Nat Biotechnol 14 (10), 1246-1251 (1996)\n",
      "   PUBMED   9631087\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (1.9 ANGSTROMS).\n",
      "REFERENCE   28 (residues 1 to 238)\n",
      "  AUTHORS   Ormo,M., Cubitt,A.B., Kallio,K., Gross,L.A., Tsien,R.Y. and\n",
      "            Remington,S.J.\n",
      "  TITLE     Crystal structure of the Aequorea victoria green fluorescent\n",
      "            protein\n",
      "  JOURNAL   Science 273 (5280), 1392-1395 (1996)\n",
      "   PUBMED   8703075\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (1.90 ANGSTROMS), AND MUTAGENESIS OF SER-65\n",
      "            AND THR-203.\n",
      "REFERENCE   29 (residues 1 to 238)\n",
      "  AUTHORS   Palm,G.J., Zdanov,A., Gaitanaris,G.A., Stauber,R., Pavlakis,G.N.\n",
      "            and Wlodawer,A.\n",
      "  TITLE     The structural basis for spectral variations in green fluorescent\n",
      "            protein\n",
      "  JOURNAL   Nat Struct Biol 4 (5), 361-365 (1997)\n",
      "   PUBMED   9145105\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.00 ANGSTROMS), MUTAGENESIS OF PHE-64;\n",
      "            TYR-66; SER-72; ASN-149; MET-153 AND ILE-167, BIOPHYSICOCHEMICAL\n",
      "            PROPERTIES, AND BIOTECHNOLOGY.\n",
      "REFERENCE   30 (residues 1 to 238)\n",
      "  AUTHORS   Wachter,R.M., Elsliger,M.A., Kallio,K., Hanson,G.T. and\n",
      "            Remington,S.J.\n",
      "  TITLE     Structural basis of spectral shifts in the yellow-emission variants\n",
      "            of green fluorescent protein\n",
      "  JOURNAL   Structure 6 (10), 1267-1277 (1998)\n",
      "   PUBMED   9782051\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.5 ANGSTROMS) OF YELLOW EMISSION MUTANT\n",
      "            TYR-203/GLY-65/LEU-68/ALA-72, MUTAGENESIS OF SER-65; VAL-68; SER-72\n",
      "            AND THR-203, BIOPHYSICOCHEMICAL PROPERTIES, BIOTECHNOLOGY, AND\n",
      "            SUBUNIT.\n",
      "REFERENCE   31 (residues 1 to 238)\n",
      "  AUTHORS   Elsliger,M.A., Wachter,R.M., Hanson,G.T., Kallio,K. and\n",
      "            Remington,S.J.\n",
      "  TITLE     Structural and spectral response of green fluorescent protein\n",
      "            variants to changes in pH\n",
      "  JOURNAL   Biochemistry 38 (17), 5296-5301 (1999)\n",
      "   PUBMED   10220315\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.0 ANGSTROMS), AND BIOPHYSICOCHEMICAL\n",
      "            PROPERTIES.\n",
      "REFERENCE   32 (residues 1 to 238)\n",
      "  AUTHORS   Griesbeck,O., Baird,G.S., Campbell,R.E., Zacharias,D.A. and\n",
      "            Tsien,R.Y.\n",
      "  TITLE     Reducing the environmental sensitivity of yellow fluorescent\n",
      "            protein. Mechanism and applications\n",
      "  JOURNAL   J Biol Chem 276 (31), 29188-29194 (2001)\n",
      "   PUBMED   11387331\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.20 ANGSTROMS) OF 2-238 OF MUTANT\n",
      "            CITRININE, AND MUTAGENESIS OF VAL-68; GLN-69; SER-72 AND THR-203.\n",
      "REFERENCE   33 (residues 1 to 238)\n",
      "  AUTHORS   Rekas,A., Alattia,J.R., Nagai,T., Miyawaki,A. and Ikura,M.\n",
      "  TITLE     Crystal structure of venus, a yellow fluorescent protein with\n",
      "            improved maturation and reduced environmental sensitivity\n",
      "  JOURNAL   J Biol Chem 277 (52), 50573-50578 (2002)\n",
      "   PUBMED   12370172\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.20 ANGSTROMS) OF 2-238 OF VENUS, AND\n",
      "            MUTAGENESIS OF PHE-46; PHE-64; SER-65; VAL-68; SER-72; MET-153;\n",
      "            VAL-163; SER-175 AND THR-203.\n",
      "REFERENCE   34 (residues 1 to 238)\n",
      "  AUTHORS   Malo,G.D., Pouwels,L.J., Wang,M., Weichsel,A., Montfort,W.R.,\n",
      "            Rizzo,M.A., Piston,D.W. and Wachter,R.M.\n",
      "  TITLE     X-ray structure of Cerulean GFP: a tryptophan-based chromophore\n",
      "            useful for fluorescence lifetime imaging\n",
      "  JOURNAL   Biochemistry 46 (35), 9865-9873 (2007)\n",
      "   PUBMED   17685554\n",
      "  REMARK    X-RAY CRYSTALLOGRAPHY (2.00 ANGSTROMS), MUTAGENESIS OF\n",
      "            65-SER-TYR-66, BIOPHYSICOCHEMICAL PROPERTIES, AND BIOTECHNOLOGY.\n",
      "COMMENT     On or before Jan 29, 2008 this sequence version replaced\n",
      "            gi:74820441, gi:7428731.\n",
      "            [FUNCTION] Energy-transfer acceptor. Its role is to transduce the\n",
      "            blue chemiluminescence of the protein aequorin into green\n",
      "            fluorescent light by energy transfer. Fluoresces in vivo upon\n",
      "            receiving energy from the Ca(2+)-activated photoprotein aequorin.\n",
      "            [BIOPHYSICOCHEMICAL PROPERTIES] Absorption: Abs(max)=395 nm;\n",
      "            Note=Exhibits a smaller absorbance peak at 470 nm. The fluorescence\n",
      "            emission spectrum peaks at 507-510 nm with a shoulder at 545 nm\n",
      "            (PubMed:8137953, PubMed:9154981). The exact value of the emission\n",
      "            maximum depends on the environment of the chromophore\n",
      "            (PubMed:10220315). As a consequence, mutant versions have been\n",
      "            designed that have substantially shifted emission spectra,\n",
      "            including yellow-emission variants (YFP), blue and cerulean\n",
      "            fluorescing proteins (PubMed:17685554, PubMed:9145105,\n",
      "            PubMed:9782051). {ECO:0000269|PubMed:10220315,\n",
      "            ECO:0000269|PubMed:17685554, ECO:0000269|PubMed:8137953,\n",
      "            ECO:0000269|PubMed:9145105, ECO:0000269|PubMed:9154981,\n",
      "            ECO:0000269|PubMed:9782051}.\n",
      "            [SUBUNIT] Monomer. {ECO:0000269|PubMed:9782051}.\n",
      "            [TISSUE SPECIFICITY] Photocytes.\n",
      "            [PTM] Contains a chromophore consisting of modified amino acid\n",
      "            residues. The chromophore is formed by autocatalytic backbone\n",
      "            condensation between Ser-65 and Gly-67, and oxidation of Tyr-66 to\n",
      "            didehydrotyrosine. Maturation of the chromophore requires nothing\n",
      "            other than molecular oxygen. {ECO:0000269|PubMed:8448132}.\n",
      "            [BIOTECHNOLOGY] Green fluorescent protein has been engineered to\n",
      "            produce a vast number of variously colored mutants, fusion\n",
      "            proteins, and biosensors. Green fluorescent protein can be mutated\n",
      "            to emit at different wavelengths such as blue for BFP (when Tyr-66\n",
      "            is replaced by His), cyan for CFP (when Tyr-66 is replaced by Trp),\n",
      "            and yellow for YFP (when Thr-203 is replaced by Tyr). Further\n",
      "            generation of mutants led to more stable proteins (at 37 degrees\n",
      "            Celsius for example) with brighter fluorescence and longer\n",
      "            fluorescence lifetimes. Fluorescent proteins and their mutated\n",
      "            allelic forms have become a useful and ubiquitous tool for making\n",
      "            chimeric proteins, where they function as a fluorescent protein\n",
      "            tag. Typically they tolerate N- and C-terminal fusion to a broad\n",
      "            variety of proteins. They have been expressed in most known cell\n",
      "            types and are used as a noninvasive fluorescent marker in living\n",
      "            cells and organisms. They enable a wide range of applications where\n",
      "            they have functioned as a cell lineage tracer, reporter of gene\n",
      "            expression, or as a measure of protein-protein interactions\n",
      "            (PubMed:17685514, PubMed:17685554, PubMed:8578587, PubMed:8707053,\n",
      "            PubMed:9145105, PubMed:9154981, PubMed:9759496, PubMed:9782051).\n",
      "            Can also be used as a molecular thermometer, allowing accurate\n",
      "            temperature measurements in fluids. The measurement process relies\n",
      "            on the detection of the blinking of GFP using fluorescence\n",
      "            correlation spectroscopy (PubMed:17685514).\n",
      "            {ECO:0000269|PubMed:17685514, ECO:0000269|PubMed:17685554,\n",
      "            ECO:0000269|PubMed:8578587, ECO:0000269|PubMed:8707053,\n",
      "            ECO:0000269|PubMed:9145105, ECO:0000269|PubMed:9154981,\n",
      "            ECO:0000269|PubMed:9759496, ECO:0000269|PubMed:9782051}.\n",
      "            [SIMILARITY] Belongs to the GFP family. {ECO:0000305}.\n",
      "            [WEB RESOURCE] Name=Protein Spotlight; Note=The greenest of us all\n",
      "            - Issue 11 of June 2001;\n",
      "            URL='https://web.expasy.org/spotlight/back_issues/011'.\n",
      "            [WEB RESOURCE] Name=Protein Spotlight; Note=Paint my thoughts -\n",
      "            Issue 108 of August 2009;\n",
      "            URL='https://web.expasy.org/spotlight/back_issues/108'.\n",
      "            [WEB RESOURCE] Name=Wikipedia; Note=Green fluorescent protein\n",
      "            entry;\n",
      "            URL='https://en.wikipedia.org/wiki/Green_fluorescent_protein'.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..238\n",
      "                     /organism=\"Aequorea victoria\"\n",
      "                     /db_xref=\"taxon:6100\"\n",
      "     gene            1..238\n",
      "                     /gene=\"GFP\"\n",
      "     Protein         1..238\n",
      "                     /product=\"Green fluorescent protein\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..238\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /note=\"Green fluorescent protein. /id=PRO_0000192576.\"\n",
      "     Region          1..3\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6UHQ.\"\n",
      "     Region          2\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"S -> G (in Ref. 3; CAA65278).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          5..8\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          12..22\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          18..201\n",
      "                     /region_name=\"GFP\"\n",
      "                     /note=\"Green fluorescent protein; pfam01353\"\n",
      "                     /db_xref=\"CDD:426217\"\n",
      "     Region          25..36\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          25\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"H -> Q (in Ref. 3; CAA65278).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Site            30\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->R: In mut1.28; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.76 ns; when associated with H-145. In mut2.2;\n",
      "                     shifts fluorescence lifetime from 3.03 to 1.94 ns; when\n",
      "                     associated with H-69 and H-145. In mut3.3; shifts\n",
      "                     fluorescence lifetime from 3.03 to 1.88 ns; when\n",
      "                     associated with L-46; H-69 and H-145. In EBFP1.2; shifts\n",
      "                     the excitation and emission spectra to shorter wavelengths\n",
      "                     and increases quantum yields compared to BFP; when\n",
      "                     associated with N-39; H-66; A-72; T-105; F-145; V-171;\n",
      "                     S-198 and V-206. In EBFP2.0; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with N-39;\n",
      "                     H-66; A-72; T-105; V-128; F-145; I-150; V-155; V-171;\n",
      "                     S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762,\n",
      "                     ECO:0000269|PubMed:22891786.\"\n",
      "     Region          37..39\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            39\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->N: In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     H-66; A-72; T-105; F-145; V-171; S-198 and V-206. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; H-66; A-72; T-105;\n",
      "                     V-128; F-145; I-150; V-155; V-171; S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          41..48\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            46\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"F->L: In mut3.3; shifts fluorescence lifetime from\n",
      "                     3.03 to 1.88 ns; when associated with R-30; H-69 and\n",
      "                     H-145. In R10-3; matures to the red-emitting state with\n",
      "                     excitation and emission maxima at 555 and 585 nm,\n",
      "                     respectively; when associated with L-64; N-68; Q-162;\n",
      "                     A-163; V-167 and L-171. In Venus; leads to yellow\n",
      "                     fluorescence, improved maturation and reduced\n",
      "                     environmental sensitivity; when associated with L-64;\n",
      "                     G-65; L-68; A-72; T-153; A-163; G-175 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:16201762, ECO:0000269|PubMed:18366185.\"\n",
      "     Region          50..52\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:5LEM.\"\n",
      "     Region          57..60\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          61..64\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            64\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"F->L: In EGFP; increases fluorescence at warmer\n",
      "                     temperatures such as 37 degrees Celsius; when associated\n",
      "                     with T-65. In EBFP; gives rise to variants with blue\n",
      "                     fluorescence; when associated with T-65 and H-66. In ECFP;\n",
      "                     leads to cyan fluorescence, folds faster and more\n",
      "                     efficiently at 37 degrees Celsius and has superior\n",
      "                     solubility and brightness; when associated with T-65;\n",
      "                     W-66; I-146; T-153 and A-163. In Cerulean; leads to\n",
      "                     improved quantum yield, a higher extinction coefficient\n",
      "                     and is 2.5-fold brighterH than ECFP; when associated with\n",
      "                     T-65; W-66; A-72; A-145; I-146; D-148; T-153 and A-163. In\n",
      "                     R10-3; matures to the red-emitting state with excitation\n",
      "                     and emission maxima at 555 and 585 nm, respectively; when\n",
      "                     associated with L-46; N-68; Q-162; A-163; V-167 and L-171.\n",
      "                     In GFPmut 1; red-shifts by about 100 nm the excitation\n",
      "                     maxima, permitting efficient excitation at 488 nm and\n",
      "                     increases fluorescence; when associated with T-65. In\n",
      "                     VisGreen; leads to brighter fluorescence; when associated\n",
      "                     with T-65; A-72; K-149 and T-167. In Venus; leads to\n",
      "                     yellow fluorescence, improved maturation and reduced\n",
      "                     environmental sensitivity; when associated with L-46;\n",
      "                     G-65; L-68; A-72; T-153; A-163; G-175 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:11776264, ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:14990965, ECO:0000269|PubMed:15065907,\n",
      "                     ECO:0000269|PubMed:17658219, ECO:0000269|PubMed:18366185,\n",
      "                     ECO:0000269|PubMed:8707053, ECO:0000269|PubMed:9145105,\n",
      "                     ECO:0000269|PubMed:9370472.\"\n",
      "     Site            64\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"F->M: In RSGFP4; increases fluorescence and shifts\n",
      "                     the major exitation peak to 489-490 nm; when associated\n",
      "                     with G-65 and L-69. /evidence=ECO:0000269|PubMed:9634755.\"\n",
      "     Region          65..67\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:1QYO.\"\n",
      "     Bond            bond(65,67)\n",
      "                     /bond_type=\"xlink\"\n",
      "                     /note=\"5-imidazolinone (Ser-Gly).\n",
      "                     /evidence=ECO:0000269|PubMed:8448132.\"\n",
      "     Site            65..66\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"SY->TW: Gives rise to variants with cerulean\n",
      "                     fluorsecence. /evidence=ECO:0000269|PubMed:17685554.\"\n",
      "     Site            65\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->A: In GFPmut 2; red-shifts by about 100 nm the\n",
      "                     excitation maxima, permitting efficient excitation at 488\n",
      "                     nm and increases fluorescence; when associated with L-68\n",
      "                     and A-72. /evidence=ECO:0000269|PubMed:8707053.\"\n",
      "     Site            65\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->G: In EYFP; leads to yellow fluorescence, folds\n",
      "                     faster and more efficiently at 37 degrees Celsius and has\n",
      "                     superior solubility and brightness; when associated with\n",
      "                     L-68; A-72 and Y-203. In GFPmut 3; highly fluorescent\n",
      "                     mutant when excited at 488 nm; when associated with A-72.\n",
      "                     Highly fluorescent; when associated with Y-203; L-68 and\n",
      "                     A-72. In RSGFP4; increases fluorescence and shifts the\n",
      "                     major exitation peak to 489-490 nm; when associated with\n",
      "                     M-64 and L-69. In YFP 10C; shifts the major emission and\n",
      "                     exitation peak up to 20 nm; when associated with L-68;\n",
      "                     A-72 and Y-203. In Topaz; shifts the major emission and\n",
      "                     exitation peak up to 20 nm; when associated with A-72;\n",
      "                     R-79 and Y-203. In Venus; leads to yellow fluorescence,\n",
      "                     improved maturation and reduced environmental sensitivity;\n",
      "                     when associated with L-46; L-64; L-68; A-72; T-153; A-163;\n",
      "                     G-175 and Y-203. /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:12370172, ECO:0000269|PubMed:8707053,\n",
      "                     ECO:0000269|PubMed:9634755, ECO:0000269|PubMed:9759496,\n",
      "                     ECO:0000269|PubMed:9782051.\"\n",
      "     Site            65\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->T: Increases fluorescence, photostability and\n",
      "                     shifts the major exitation peak to 488 nm. In EGFP;\n",
      "                     increases fluorescence at warmer temperatures such as 37\n",
      "                     degrees Celsius; when associated with L-64. In EBFP; gives\n",
      "                     rise to variants with blue fluorescence; when associated\n",
      "                     with L-64 and H-66. In GFPmut 1; red-shifts by about 100\n",
      "                     nm the excitation maxima, permitting efficient excitation\n",
      "                     at 488 nm and increases fluorescence; when associated with\n",
      "                     T-65. In ECFP; leads to cyan fluorescence, folds faster\n",
      "                     and more efficiently at 37 degrees Celsius and has\n",
      "                     superior solubility and brightness; when associated with\n",
      "                     L-64; W-66; I-146; T-153 and A-163. In Cerulean; leads to\n",
      "                     improved quantum yield, a higher extinction coefficient\n",
      "                     and is 2.5-fold brighter than ECFP; when associated with\n",
      "                     L-64; W-66; A-72; A-145; I-146; D-148; T-153 and A-163. In\n",
      "                     Esmerald; leads to brighter fluorescence; when associated\n",
      "                     with A-72; K-149; T-153 and T-167. In VisGreen; leads to\n",
      "                     brighter fluorescence; when associated with L-64; A-72;\n",
      "                     K-149 and T-167. /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:11776264, ECO:0000269|PubMed:14990965,\n",
      "                     ECO:0000269|PubMed:15065907, ECO:0000269|PubMed:17658219,\n",
      "                     ECO:0000269|PubMed:8703075, ECO:0000269|PubMed:8707053,\n",
      "                     ECO:0000269|PubMed:9370472.\"\n",
      "     Site            66\n",
      "                     /site_type=\"modified\"\n",
      "                     /note=\"(Z)-2,3-didehydrotyrosine.\n",
      "                     /evidence=ECO:0000269|PubMed:8448132.\"\n",
      "     Site            66\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->H: In BFP; shifts the excitation and emission\n",
      "                     spectra to shorter wavelengths. In EBFP; gives rise to\n",
      "                     variants with blue fluorescence; when associated with L-64\n",
      "                     and T-65. In Azurite; shifts the excitation and emission\n",
      "                     spectra to shorter wavelengths and increases quantum\n",
      "                     yields compared to BFP; when associated with R-80; F-145;\n",
      "                     I-150 and R-224. In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; A-72; T-105; F-145; V-171; S-198 and V-206. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; A-72; T-105;\n",
      "                     V-128; F-145; I-150; V-155; V-171; S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786,\n",
      "                     ECO:0000269|PubMed:8578587, ECO:0000269|PubMed:9145105,\n",
      "                     ECO:0000269|PubMed:9370472.\"\n",
      "     Site            66\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->T,F: Shifts the excitation and emission spectra\n",
      "                     to shorter wavelengths.\n",
      "                     /evidence=ECO:0000269|PubMed:8578587.\"\n",
      "     Site            66\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->W: In W; leads to excitation and emission\n",
      "                     wavelengths intermediate between tyrosine and histidine\n",
      "                     but is only weakly fluorescent. In ECFP; leads to cyan\n",
      "                     fluorescence, folds faster and more efficiently at 37\n",
      "                     degrees Celsius and has superior solubility and\n",
      "                     brightness; when associated with L-64; T-65; I-146; T-153\n",
      "                     and A-163. In Cerulean; leads to improved quantum yield, a\n",
      "                     higher extinction coefficient and is 2.5-fold brighter\n",
      "                     than ECFP; when associated with L-64; T-65; A-72; A-145;\n",
      "                     I-146; D-148; T-153 and A-163.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:14990965, ECO:0000269|PubMed:15065907,\n",
      "                     ECO:0000269|PubMed:7809066.\"\n",
      "     Site            68\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->L: In EYFP; leads to yellow fluorescence, folds\n",
      "                     faster and more efficiently at 37 degrees Celsius and has\n",
      "                     superior solubility and brightness; when associated with\n",
      "                     G-65; A-72 and Y-203. In GFPmut 2; red-shifts by about 100\n",
      "                     nm the excitation maxima, permitting efficient excitation\n",
      "                     at 488 nm and increases fluorescence; when associated with\n",
      "                     A-65 and A-72. In Citrinine; leads to excitation and\n",
      "                     emission peaks of 516 and 529 nm, respectively; when\n",
      "                     associated with M-69; A-72 and Y-203. In YFP 10C; shifts\n",
      "                     the major emission and exitation peak up to 20 nm; when\n",
      "                     associated with G-65; A-72 and Y-203. In Venus; leads to\n",
      "                     yellow fluorescence, improved maturation and reduced\n",
      "                     environmental sensitivity; when associated with L-46;\n",
      "                     L-64; G-65; A-72; T-153; A-163; G-175 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:11387331, ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:8707053, ECO:0000269|PubMed:9759496,\n",
      "                     ECO:0000269|PubMed:9782051.\"\n",
      "     Site            68\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->N: In R10-3; matures to the red-emitting state\n",
      "                     with excitation and emission maxima at 555 and 585 nm,\n",
      "                     respectively; when associated with L-46; L-64; Q-162;\n",
      "                     A-163; V-167 and L-171.\n",
      "                     /evidence=ECO:0000269|PubMed:18366185.\"\n",
      "     Region          69..71\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            69\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Q->H: In P4; leads to no detectable fluorescence.\n",
      "                     In mut2.2; shifts fluorescence lifetime from 3.03 to 1.94\n",
      "                     ns; when associated with R-30 and H-145. In mut3.3; shifts\n",
      "                     fluorescence lifetime from 3.03 to 1.88 ns; when\n",
      "                     associated with R-30; L-46 and H-145.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762,\n",
      "                     ECO:0000269|PubMed:7809066.\"\n",
      "     Site            69\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Q->L: In RSGFP4; increases fluorescence and shifts\n",
      "                     the major exitation peak to 489-490 nm; when associated\n",
      "                     with M-64 and G-65. /evidence=ECO:0000269|PubMed:9634755.\"\n",
      "     Site            69\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Q->M: In Citrinine; leads to excitation and\n",
      "                     emission peaks of 516 and 529 nm, respectively; when\n",
      "                     associated with L-68; A-72 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:11387331.\"\n",
      "     Region          72..74\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6OA8.\"\n",
      "     Site            72\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->A: Increases fluorescence at warmer temperatures\n",
      "                     such as 37 degrees Celsius. In GFPmut 3; highly\n",
      "                     fluorescent mutant when excited at 488 nm; when associated\n",
      "                     with G-65. In EYFP; leads to yellow fluorescence, folds\n",
      "                     faster and more efficiently at 37 degrees Celsius and has\n",
      "                     superior solubility and brightness; when associated with\n",
      "                     G-65; L-68 and Y-203. In GFPmut 2; red-shifts by about 100\n",
      "                     nm the excitation maxima, permitting efficient excitation\n",
      "                     at 488 nm and increases fluorescence; when associated with\n",
      "                     A-65 and L-68. In Citrinine; leads to excitation and\n",
      "                     emission peaks of 516 and 529 nm, respectively; when\n",
      "                     associated with L-68; M-69 and Y-203. In YFP 10C; shifts\n",
      "                     the major emission and exitation peak up to 20 nm; when\n",
      "                     associated with G-65; L-68; and Y-203. In Topaz; shifts\n",
      "                     the major emission and exitation peak up to 20 nm; when\n",
      "                     associated with G-65; R-79 and Y-203. In Cerulean; leads\n",
      "                     to improved quantum yield, a higher extinction coefficient\n",
      "                     and is 2.5-fold brighter than ECFP; when associated with\n",
      "                     L-64; T-65; W-66; A-145; I-146; D-148; T-153 and A-163. In\n",
      "                     Sapphire/H9-40; exhibits a huge Stoke's shift, with an\n",
      "                     excitation peak at 399 nm and an emission peak at 511 nm;\n",
      "                     when associated with F-145 and I-203. In EBFP1.2; shifts\n",
      "                     the excitation and emission spectra to shorter wavelengths\n",
      "                     and increases quantum yields compared to BFP; when\n",
      "                     associated with R-30; N-39; H-66; T-105; F-145; V-171;\n",
      "                     S-198 and V-206. In EBFP2.0; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; T-105; V-128; F-145; I-150; V-155; V-171;\n",
      "                     S-198; V-206 and V-224. In Esmerald; leads to brighter\n",
      "                     fluorescence; when associated with T-65; K-149; T-153 and\n",
      "                     T-167. In VisGreen; leads to brighter fluorescence; when\n",
      "                     associated with L-64; T-65; K-149 and T-167. In Venus;\n",
      "                     leads to yellow fluorescence, improved maturation and\n",
      "                     reduced environmental sensitivity; when associated with\n",
      "                     L-46; L-64; G-65; L-68; T-153; A-163; G-175 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:11387331, ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:12769828, ECO:0000269|PubMed:14990965,\n",
      "                     ECO:0000269|PubMed:17658219, ECO:0000269|PubMed:22891786,\n",
      "                     ECO:0000269|PubMed:8707053, ECO:0000269|PubMed:9145105,\n",
      "                     ECO:0000269|PubMed:9759496, ECO:0000269|PubMed:9782051.\"\n",
      "     Region          76..81\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            79\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"K->R: In Topaz; shifts the major emission and\n",
      "                     exitation peak up to 20 nm; when associated with G-65;\n",
      "                     A-72 and Y-203. /evidence=ECO:0000269|PubMed:9759496.\"\n",
      "     Region          80\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"Q -> R (in Ref. 3; CAA65278).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Site            80\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Q->R: In Azurite; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with H-66;\n",
      "                     F-145; I-150 and R-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          83..86\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          87..90\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          92..100\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            99\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"F->S: In alphaGFP/cycle 3 GFP; improves folding at\n",
      "                     37 degrees Celsius, reduces aggregation at high\n",
      "                     concentrations, and increases the diffusibility of the\n",
      "                     protein inside cells; when associated with T-153 and\n",
      "                     A-163. /evidence=ECO:0000269|PubMed:9370472,\n",
      "                     ECO:0000269|PubMed:9630892, ECO:0000269|PubMed:9631088.\"\n",
      "     Region          100\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"F -> Y. /evidence=ECO:0000269|PubMed:1347277.\"\n",
      "     Region          101..103\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6MWQ.\"\n",
      "     Site            103\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"D->E: In mut1.27; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.85 ns; when associated with H-145.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Region          105..115\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            105\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"N->T: In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; F-145; V-171; S-198 and V-206. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; H-66; A-72;\n",
      "                     V-128; F-145; I-150; V-155; V-171; S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          108\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"T -> S. /evidence=ECO:0000269|PubMed:1347277.\"\n",
      "     Region          118..129\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            128\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"I->V: In EBFP2.0; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; F-145; I-150; V-155; V-171;\n",
      "                     S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          132..134\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:1EMK.\"\n",
      "     Region          135..139\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          141..147\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:4ZF5.\"\n",
      "     Region          141\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"L -> M. /evidence=ECO:0000269|PubMed:1347277.\"\n",
      "     Site            145\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->A: In Cerulean; leads to improved quantum yield,\n",
      "                     a higher extinction coefficient and is 2.5-fold brighter\n",
      "                     than ECFP; when associated with L-64; T-65; W-66; A-72;\n",
      "                     I-146; D-148; T-153 and A-163.\n",
      "                     /evidence=ECO:0000269|PubMed:14990965.\"\n",
      "     Site            145\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->C: In mut1.9; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.74 ns. /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Site            145\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->F: In Sapphire/H9-40; exhibits a huge Stoke's\n",
      "                     shift, with an excitation peak at 399 nm and an emission\n",
      "                     peak at 511 nm; when associated with A-72 and I-203. In\n",
      "                     Azurite; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with H-66; R-80; I-150 and R-224.\n",
      "                     In EBFP1.2; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; H-66; A-72;\n",
      "                     T-105; V-171; S-198 and V-206. In EBFP2.0; shifts the\n",
      "                     excitation and emission spectra to shorter wavelengths and\n",
      "                     increases quantum yields compared to BFP; when associated\n",
      "                     with R-30; N-39; H-66; A-72; T-105; V-128; I-150; V-155;\n",
      "                     V-171; S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:12769828,\n",
      "                     ECO:0000269|PubMed:22891786.\"\n",
      "     Site            145\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"Y->H: In mut1.3; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.78 ns. In mut1.5; shifts fluorescence lifetime\n",
      "                     from 3.03 to 2.72 ns; when associated with A-193. In\n",
      "                     mut1.27; shifts fluorescence lifetime from 3.03 to 2.85\n",
      "                     ns; when associated with E-103. In mut1.28; shifts\n",
      "                     fluorescence lifetime from 3.03 to 2.76 ns; when\n",
      "                     associated with R-30. In mut2.1; shifts fluorescence\n",
      "                     lifetime from 3.03 to 2.50 ns; when associated with A-176\n",
      "                     and I-198. In mut2.2; shifts fluorescence lifetime from\n",
      "                     3.03 to 1.94 ns; when associated with R-30 and H-69. In\n",
      "                     mut3.3; shifts fluorescence lifetime from 3.03 to 1.88 ns;\n",
      "                     when associated with R-30; L-46 and H-69.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Site            146\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"N->I: In ECFP; leads to cyan fluorescence, folds\n",
      "                     faster and more efficiently at 37 degrees Celsius and has\n",
      "                     superior solubility and brightness; when associated with\n",
      "                     L-64; T-65; W-66; T-153 and A-163. In Cerulean; leads to\n",
      "                     improved quantum yield, a higher extinction coefficient\n",
      "                     and is 2.5-fold brighter than ECFP; when associated with\n",
      "                     L-64; T-65; W-66; A-72; A-145; D-148; T-153 and A-163.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:14990965, ECO:0000269|PubMed:15065907.\"\n",
      "     Site            147\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->P: Increases fluorescence at warmer temperatures\n",
      "                     such as 37 degrees Celsius.\n",
      "                     /evidence=ECO:0000269|PubMed:9125154.\"\n",
      "     Region          148..154\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            148\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"H->D: In Cerulean; leads to improved quantum yield,\n",
      "                     a higher extinction coefficient and is 2.5-fold brighter\n",
      "                     than ECFP; when associated with L-64; T-65; W-66; A-72;\n",
      "                     A-145; I-146; T-153 and A-163.\n",
      "                     /evidence=ECO:0000269|PubMed:14990965.\"\n",
      "     Site            149\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"N->K: Increases fluorescence at warmer temperatures\n",
      "                     such as 37 degrees Celsius. In Esmerald; leads to brighter\n",
      "                     fluorescence; when associated with T-65; A-72; T-153 and\n",
      "                     T-167. In VisGreen; leads to brighter fluorescence; when\n",
      "                     associated with L-64; T-65; A-72 and T-167.\n",
      "                     /evidence=ECO:0000269|PubMed:17658219,\n",
      "                     ECO:0000269|PubMed:9145105.\"\n",
      "     Site            150\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->I: In Azurite; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with H-66;\n",
      "                     R-80; F-145 and R-224. In EBFP2.0; shifts the excitation\n",
      "                     and emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; V-128; F-145; V-155; V-171;\n",
      "                     S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Site            153\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"M->T: Increases fluorescence at warmer temperatures\n",
      "                     such as 37 degrees Celsius. In alphaGFP/cycle 3 GFP;\n",
      "                     improves folding at 37 degrees Celsius, reduces\n",
      "                     aggregation at high concentrations, and increases the\n",
      "                     diffusibility of the protein inside cells; when associated\n",
      "                     with S-99 and A-163. In ECFP; leads to cyan fluorescence,\n",
      "                     folds faster and more efficiently at 37 degrees Celsius\n",
      "                     and has superior solubility and brightness; when\n",
      "                     associated with L-64; T-65; W-66; I-146 and A-163. In\n",
      "                     Cerulean; leads to improved quantum yield, a higher\n",
      "                     extinction coefficient and is 2.5-fold brighter than ECFP;\n",
      "                     when associated with L-64; T-65; W-66; A-72; A-145; I-146;\n",
      "                     D-148 and A-163. In Esmerald; leads to brighter\n",
      "                     fluorescence; when associated with T-65; A-72; K-149 and\n",
      "                     T-167. In Venus; leads to yellow fluorescence, improved\n",
      "                     maturation and reduced environmental sensitivity; when\n",
      "                     associated with L-46; L-64; G-65; L-68; A-72; A-163; G-175\n",
      "                     and Y-203. /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:12370172, ECO:0000269|PubMed:14990965,\n",
      "                     ECO:0000269|PubMed:15065907, ECO:0000269|PubMed:17658219,\n",
      "                     ECO:0000269|PubMed:9145105, ECO:0000269|PubMed:9370472,\n",
      "                     ECO:0000269|PubMed:9630892, ECO:0000269|PubMed:9631088.\"\n",
      "     Site            155\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"D->V: In EBFP2.0; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; V-128; F-145; I-150; V-171;\n",
      "                     S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          157\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"Q -> P (in Ref. 2; AAA58246).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          158..171\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            162\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"K->Q: In R10-3; matures to the red-emitting state\n",
      "                     with excitation and emission maxima at 555 and 585 nm,\n",
      "                     respectively; when associated with L-46; L-64; N-68;\n",
      "                     A-163; V-167 and L-171.\n",
      "                     /evidence=ECO:0000269|PubMed:18366185.\"\n",
      "     Site            163\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->A: In GFPB; leads to enhanced fluorescence at 37\n",
      "                     degrees Celsius. In GFPA; leads to even higher\n",
      "                     fluorescence at 37 degrees Celsius than GFPA;\n",
      "                     whenassociated with G-175. In alphaGFP/cycle 3 GFP;\n",
      "                     improves folding at 37 degrees Celsius, reduces\n",
      "                     aggregation at high concentrations, and increases the\n",
      "                     diffusibility of the protein inside cells; when associated\n",
      "                     with S-99 and T-153. In R10-3; matures to the red-emitting\n",
      "                     state with excitation and emission maxima at 555 and 585\n",
      "                     nm, respectively; when associated with L-46; L-64; N-68;\n",
      "                     Q-162; V-167 and L-171. In ECFP; leads to cyan\n",
      "                     fluorescence, folds faster and more efficiently at 37\n",
      "                     degrees Celsius and has superior solubility and\n",
      "                     brightness; when associated with L-64; T-65; W-66; I-146\n",
      "                     and T-153. In Cerulean; leads to improved quantum yield, a\n",
      "                     higher extinction coefficient and is 2.5-fold brighter\n",
      "                     than ECFP; when associated with L-64; T-65; W-66; A-72;\n",
      "                     A-145; I-146; D-148 and T-153. In Venus; leads to yellow\n",
      "                     fluorescence, improved maturation and reduced\n",
      "                     environmental sensitivity; when associated with L-46;\n",
      "                     L-64; G-65; L-68; A-72; T-153; G-175 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:12370172, ECO:0000269|PubMed:14990965,\n",
      "                     ECO:0000269|PubMed:15065907, ECO:0000269|PubMed:18366185,\n",
      "                     ECO:0000269|PubMed:8994830, ECO:0000269|PubMed:9370472,\n",
      "                     ECO:0000269|PubMed:9630892, ECO:0000269|PubMed:9631088.\"\n",
      "     Site            167\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"I->T: In P11; increases fluorescence at 475 nm\n",
      "                     excitation and at warmer temperatures such as 37 degrees\n",
      "                     Celsius. In Esmerald; leads to brighter fluorescence; when\n",
      "                     associated with T-65; A-72; K-149 and T-153. In VisGreen;\n",
      "                     leads to brighter fluorescence; when associated with L-64;\n",
      "                     T-65; A-72 and K-149.\n",
      "                     /evidence=ECO:0000269|PubMed:17658219,\n",
      "                     ECO:0000269|PubMed:7809066, ECO:0000269|PubMed:9145105.\"\n",
      "     Site            167\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"I->V: In P9; increases fluorescence at 475 nm\n",
      "                     excitation. In R10-3; matures to the red-emitting state\n",
      "                     with excitation and emission maxima at 555 and 585 nm,\n",
      "                     respectively; when associated with L-46; L-64; N-68;\n",
      "                     Q-162; A-163 and L-171.\n",
      "                     /evidence=ECO:0000269|PubMed:18366185,\n",
      "                     ECO:0000269|PubMed:7809066.\"\n",
      "     Site            171\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"I->L: In R10-3; matures to the red-emitting state\n",
      "                     with excitation and emission maxima at 555 and 585 nm,\n",
      "                     respectively; when associated with L-46; L-64; N-68;\n",
      "                     Q-162; A-163 and V-167.\n",
      "                     /evidence=ECO:0000269|PubMed:18366185.\"\n",
      "     Site            171\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"I->V: In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; F-145; S-198 and V-206. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; H-66; A-72;\n",
      "                     T-105; V-128; F-145; I-150; V-155; S-198; V-206 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          172..174\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:4OGS.\"\n",
      "     Region          172\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"E -> K (in Ref. 2; AAA58246).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Site            175\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->G: In GFPA; leads to enhanced fluorescence at 37\n",
      "                     degrees Celsius; when associated with A-163. In Venus;\n",
      "                     leads to yellow fluorescence, improved maturation and\n",
      "                     reduced environmental sensitivity; when associated with\n",
      "                     L-46; L-64; G-65; L-68; A-72; T-153; A-163 and Y-203.\n",
      "                     /evidence=ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:8994830.\"\n",
      "     Region          176..191\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            176\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->A: In mut2.1; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.50 ns; when associated with H-145 and I-198.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Site            193\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->A: In mut1.5; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.72 ns; when associated with H-145.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Region          198..208\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Site            198\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"N->I: In mut2.1; shifts fluorescence lifetime from\n",
      "                     3.03 to 2.50 ns; when associated with H-145 and A-176.\n",
      "                     /evidence=ECO:0000269|PubMed:16201762.\"\n",
      "     Site            198\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"N->S: In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; F-145; V-171 and V-206. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; H-66; A-72;\n",
      "                     T-105; V-128; F-145; I-150; V-155; V-171; S-198 and V-224.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Site            202\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"S->F: In H9; increases fluorescence at 395 nm\n",
      "                     excitation; when associated with I-203.\n",
      "                     /evidence=ECO:0000269|PubMed:7809066.\"\n",
      "     Site            203\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"T->H,W,F: Results in significantly red-shifted\n",
      "                     excitation and emission maxima.\n",
      "                     /evidence=ECO:0000269|PubMed:8703075.\"\n",
      "     Site            203\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"T->I: Suppresses the 475 nm excitation peak,\n",
      "                     leaving only the shorter wavelength peak at 399 nm. In H9;\n",
      "                     increases fluorescence at 395 nm excitation; when\n",
      "                     associated with I-203. In Sapphire/H9-40; exhibits a huge\n",
      "                     Stoke's shift, with an excitation peak at 399 nm and an\n",
      "                     emission peak at 511 nm; when associated with A-72 and\n",
      "                     F-145. /evidence=ECO:0000269|PubMed:12769828,\n",
      "                     ECO:0000269|PubMed:7796912, ECO:0000269|PubMed:7809066.\"\n",
      "     Site            203\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"T->Y: Gives rise to yellow-emission variants. In\n",
      "                     EYFP; leads to yellow fluorescence, folds faster and more\n",
      "                     efficiently at 37 degrees Celsius and has superior\n",
      "                     solubility and brightness; when associated with G-65; L-68\n",
      "                     and A-72. In Citrinine; leads to excitation and emission\n",
      "                     peaks of 516 and 529 nm, respectively; when associated\n",
      "                     with L-68; M-69 and A-72. In YFP 10C; shifts the major\n",
      "                     emission and exitation peak up to 20 nm; when associated\n",
      "                     with G-65; L-68 and A-72. In Topaz; shifts the major\n",
      "                     emission and exitation peak up to 20 nm; when associated\n",
      "                     with G-65; A-72 and R-79. In Venus; leads to yellow\n",
      "                     fluorescence, improved maturation and reduced\n",
      "                     environmental sensitivity; when associated with L-46;\n",
      "                     L-64; G-65; L-68; A-72; T-153; A-163 and G-175.\n",
      "                     /evidence=ECO:0000269|PubMed:11299042,\n",
      "                     ECO:0000269|PubMed:11387331, ECO:0000269|PubMed:12370172,\n",
      "                     ECO:0000269|PubMed:9759496, ECO:0000269|PubMed:9782051.\"\n",
      "     Site            206\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"A->K: Abolishes the tendency to dimerize and leads\n",
      "                     to monomeric fluorescent proteins.\n",
      "                     /evidence=ECO:0000269|PubMed:16739159.\"\n",
      "     Site            206\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"A->V: In EBFP1.2; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; F-145; V-171 and S-198. In\n",
      "                     EBFP2.0; shifts the excitation and emission spectra to\n",
      "                     shorter wavelengths and increases quantum yields compared\n",
      "                     to BFP; when associated with R-30; N-39; H-66; A-72;\n",
      "                     T-105; V-128; F-145; I-150; V-155; V-171; S-198 and\n",
      "                     V-206.\"\n",
      "     Region          209..213\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:4IK8.\"\n",
      "     Region          215..228\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6L27.\"\n",
      "     Region          219\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"V -> I. /evidence=ECO:0000269|PubMed:1347277.\"\n",
      "     Site            222\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"E->G: Suppresses the 399 nm excitation peak,\n",
      "                     leaving only the longer wavelength peak at 475 nm.\n",
      "                     /evidence=ECO:0000269|PubMed:7796912.\"\n",
      "     Site            224\n",
      "                     /site_type=\"mutagenized\"\n",
      "                     /note=\"V->R: In Azurite; shifts the excitation and\n",
      "                     emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with H-66;\n",
      "                     R-80; F-145 and I-150. In EBFP2.0; shifts the excitation\n",
      "                     and emission spectra to shorter wavelengths and increases\n",
      "                     quantum yields compared to BFP; when associated with R-30;\n",
      "                     N-39; H-66; A-72; T-105; V-128; F-145; I-150; V-155;\n",
      "                     V-171; S-198 and V-206.\n",
      "                     /evidence=ECO:0000269|PubMed:22891786.\"\n",
      "     Region          233..236\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:4GF6.\"\n",
      "ORIGIN      \n",
      "        1 mskgeelftg vvpilveldg dvnghkfsvs gegegdatyg kltlkfictt gklpvpwptl\n",
      "       61 vttfsygvqc fsrypdhmkq hdffksampe gyvqertiff kddgnyktra evkfegdtlv\n",
      "      121 nrielkgidf kedgnilghk leynynshnv yimadkqkng ikvnfkirhn iedgsvqlad\n",
      "      181 hyqqntpigd gpvllpdnhy lstqsalskd pnekrdhmvl lefvtaagit hgmdelyk\n",
      "//\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"protein\", term=\"P42212\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq\n",
    "print(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IEP of peptide MSKGEELFTGVVPILVELDGDVNGHKFSVSGEGEGDATYGKLTLKFICTTGKLPVPWPTLVTTFSYGVQCFSRYPDHMKQHDFFKSAMPEGYVQERTIFFKDDGNYKTRAEVKFEGDTLVNRIELKGIDFKEDGNILGHKLEYNYNSHNVYIMADKQKNGIKVNFKIRHNIEDGSVQLADHYQQNTPIGDGPVLLPDNHYLSTQSALSKDPNEKRDHMVLLEFVTAAGITHGMDELYK is 5.67\n",
      "It's charge at pH 7 is 1.25\n"
     ]
    }
   ],
   "source": [
    "protein = IP(protein_sequence)\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 7 is {:.2f}\".format(protein.charge_at_pH(5.52)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       IGG1_HUMAN               449 aa            linear   PRI 02-OCT-2024\n",
      "DEFINITION  RecName: Full=Immunoglobulin gamma-1 heavy chain; AltName:\n",
      "            Full=Immunoglobulin gamma-1 heavy chain NIE.\n",
      "ACCESSION   P0DOX5\n",
      "VERSION     P0DOX5.2\n",
      "DBSOURCE    UniProtKB: locus IGG1_HUMAN, accession P0DOX5;\n",
      "            class: standard.\n",
      "            created: Mar 15, 2017.\n",
      "            sequence updated: Jul 18, 2018.\n",
      "            annotation updated: Oct 2, 2024.\n",
      "            xrefs: 1N0X_H, 1N0X_K, 3PGF_H, 4R2G_D, 4R2G_J, 4R2G_N, 4R2G_Q,\n",
      "            5O4E_A, 5O4E_C, 5VJ6_H, 5VU0_A, 5VU0_B, 5VZX_E, 5VZX_H, 5VZY_H,\n",
      "            5W5L_A, 5W5L_B, 5WAV_A, 5WAV_B, 5XJE_A, 5XJE_B, 5XJF_A, 5XJF_B,\n",
      "            5XMH_A, 5XMH_B, 5Y56_A, 5Y56_B, 5YC5_A, 5YC5_B, 6APD_J, 6APD_K,\n",
      "            6APD_N, 6ARP_B, 6ARP_D, 6ARU_C, 6B70_C, 6B70_E, 6B7Z_C, 6B7Z_E,\n",
      "            6BF7_C, 6BF7_E, 6BF9_C, 6BF9_E, 6BFT_A, 6BFT_H, 6BGT_B, 6BKB_H,\n",
      "            6BKC_H, 6BZ4_A, 6BZ4_B, 6DKJ_A, 6DKJ_H, 6EAQ_A, 6EAQ_B, 6FCZ_H,\n",
      "            6FCZ_K, 6FGO_A, 6FGO_B, 6FGO_C, 6FGO_D, 6G1E_B, 6IFJ_A, 6IFJ_B,\n",
      "            6IQG_A, 6IQG_B, 6IQH_A, 6IQH_B, 6KA7_C, 6KA7_D, 6MB3_A, 6MB3_B,\n",
      "            6MB3_C, 6MB3_D, 6MB3_F, 6MB3_G, 6MB3_H, 6MB3_I, 6MB3_J, 6MSY_H,\n",
      "            6MU3_H, 6MU3_M, 6MUB_H, 6MUB_M, 6N2X_H, 6N2X_M, 6N32_H, 6N32_K,\n",
      "            6N35_H, 6N35_M, 6OGE_C, 6OKQ_A, 6OKQ_D, 6OKQ_F, 6P6D_A, 6P6D_B,\n",
      "            6UBI_A, 6UBI_D, 6UGW_A, 6UGX_A, 6UGX_B, 6UGY_A, 6UOE_H, 6V8Z_C,\n",
      "            6V8Z_I, 6V8Z_O, 6VSL_A, 6VSL_B, 6VSZ_A, 6VSZ_B, 6X3I_A, 6YSC_A,\n",
      "            6YSC_B, 6YT7_A, 6YTB_A, 6YTB_B, 7CZT_H, 7CZT_I, 7CZU_H, 7CZU_J,\n",
      "            7CZV_H, 7CZV_I, 7CZV_J, 7T17_H, 7URU_A, 7URU_B, 7X13_A, 7X13_B,\n",
      "            7X13_C, 7X13_D, 7X13_E, 7X13_F, 7X13_G, 7X13_H, 7X13_I, 7X13_J,\n",
      "            7X13_K, 7X13_L, 7XXL_A, 8A49_A, 8A49_B, 8A64_B, 8A64_C, 8DAO_E,\n",
      "            8DAO_G, 8DBZ_F, 8DBZ_H, 8DV1_D, 8DV2_D, 8ECQ_H, 8ECV_B, 8ECV_H,\n",
      "            8ECZ_B, 8ECZ_H, 8ED1_H, 8EDF_H, 8GHR_A, 8GHR_B, 8ZCK_A, 8ZCK_B,\n",
      "            8ZCL_A, 8ZCL_B, 8ZCM_A, 8ZCM_B\n",
      "            xrefs (non-sequence databases): PDBsum:1N0X, PDBsum:3PGF,\n",
      "            PDBsum:4R2G, PDBsum:5O4E, PDBsum:5VJ6, PDBsum:5VU0, PDBsum:5VZX,\n",
      "            PDBsum:5VZY, PDBsum:5W5L, PDBsum:5WAV, PDBsum:5XJE, PDBsum:5XJF,\n",
      "            PDBsum:5XMH, PDBsum:5Y56, PDBsum:5YC5, PDBsum:6APD, PDBsum:6ARP,\n",
      "            PDBsum:6ARU, PDBsum:6B70, PDBsum:6B7Z, PDBsum:6BF7, PDBsum:6BF9,\n",
      "            PDBsum:6BFT, PDBsum:6BGT, PDBsum:6BKB, PDBsum:6BKC, PDBsum:6BZ4,\n",
      "            PDBsum:6DKJ, PDBsum:6EAQ, PDBsum:6FCZ, PDBsum:6FGO, PDBsum:6G1E,\n",
      "            PDBsum:6IFJ, PDBsum:6IQG, PDBsum:6IQH, PDBsum:6KA7, PDBsum:6MB3,\n",
      "            PDBsum:6MSY, PDBsum:6MU3, PDBsum:6MUB, PDBsum:6N2X, PDBsum:6N32,\n",
      "            PDBsum:6N35, PDBsum:6OGE, PDBsum:6OKQ, PDBsum:6P6D, PDBsum:6UBI,\n",
      "            PDBsum:6UGW, PDBsum:6UGX, PDBsum:6UGY, PDBsum:6UOE, PDBsum:6V8Z,\n",
      "            PDBsum:6VSL, PDBsum:6VSZ, PDBsum:6X3I, PDBsum:6YSC, PDBsum:6YT7,\n",
      "            PDBsum:6YTB, PDBsum:7CZT, PDBsum:7CZU, PDBsum:7CZV, PDBsum:7T17,\n",
      "            PDBsum:7URU, PDBsum:7X13, PDBsum:7XXL, PDBsum:8A49, PDBsum:8A64,\n",
      "            PDBsum:8DAO, PDBsum:8DBZ, PDBsum:8DV1, PDBsum:8DV2, PDBsum:8ECQ,\n",
      "            PDBsum:8ECV, PDBsum:8ECZ, PDBsum:8ED1, PDBsum:8EDF, PDBsum:8GHR,\n",
      "            PDBsum:8ZCK, PDBsum:8ZCL, PDBsum:8ZCM, AlphaFoldDB:P0DOX5,\n",
      "            EMDB:EMD-15205, EMDB:EMD-25606, EMDB:EMD-27730, EMDB:EMD-27731,\n",
      "            EMDB:EMD-30516, EMDB:EMD-32933, EMDB:EMD-40047, EMDB:EMD-4232,\n",
      "            SMR:P0DOX5, CarbonylDB:P0DOX5, GlyConnect:2964, iPTMnet:P0DOX5,\n",
      "            PhosphoSitePlus:P0DOX5, jPOST:P0DOX5, SIGNOR:P0DOX5,\n",
      "            EvolutionaryTrace:P0DOX5, Pharos:P0DOX5, GO:0005576, GO:0019814,\n",
      "            GO:0005886, GO:0002250, CDD:cd21817, CDD:cd07696, CDD:cd05768,\n",
      "            CDD:cd04981, Gene3D:2.60.40.10, InterPro:IPR007110,\n",
      "            InterPro:IPR036179, InterPro:IPR013783, InterPro:IPR003006,\n",
      "            InterPro:IPR003597, InterPro:IPR003599, InterPro:IPR013106,\n",
      "            InterPro:IPR050380, PANTHER:PTHR23411:SF39, PANTHER:PTHR23411,\n",
      "            Pfam:PF07654, Pfam:PF07686, SMART:SM00409, SMART:SM00407,\n",
      "            SMART:SM00406, SUPFAM:SSF48726, PROSITE:PS50835, PROSITE:PS00290\n",
      "KEYWORDS    3D-structure; Adaptive immunity; Cell membrane; Direct protein\n",
      "            sequencing; Disulfide bond; Glycoprotein; Immunity; Immunoglobulin;\n",
      "            Immunoglobulin domain; Membrane; Pyrrolidone carboxylic acid;\n",
      "            Repeat; Secreted.\n",
      "SOURCE      Homo sapiens (human)\n",
      "  ORGANISM  Homo sapiens\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini;\n",
      "            Catarrhini; Hominidae; Homo.\n",
      "REFERENCE   1  (residues 1 to 449)\n",
      "  AUTHORS   Ponstingl,H. and Hilschmann,N.\n",
      "  TITLE     The rule of antibody structure. The primary structure of a\n",
      "            monoclonal IgG1 immunoglobulin (myeloma protein Nie). III. The\n",
      "            chymotryptic peptides of the H-chain, alignment of the tryptic\n",
      "            peptides and discussion of the complete structure\n",
      "  JOURNAL   Hoppe-Seyler's Z. Physiol. Chem. 357, 1571-1604 (1976)\n",
      "   PUBMED   826475\n",
      "  REMARK    PROTEIN SEQUENCE, AND PYROGLUTAMATE FORMATION AT GLN-1.\n",
      "REFERENCE   2  (residues 1 to 449)\n",
      "  AUTHORS   Dreker,L., Schwarz,J., Reichel,W. and Hilschmann,N.\n",
      "  TITLE     Rule of antibody structure. The primary structure of a monoclonal\n",
      "            IgG1 immunoglobulin (myeloma protein Nie), I: purification and\n",
      "            characterization of the protein, the L- and H-chains, the cyanogen\n",
      "            bromide cleavage products, and the disulfide bridges\n",
      "  JOURNAL   Hoppe-Seyler's Z. Physiol. Chem. 357, 1515-1540 (1976)\n",
      "   PUBMED   1002129\n",
      "  REMARK    DISULFIDE BOND.\n",
      "REFERENCE   3  (residues 1 to 449)\n",
      "  AUTHORS   Teng,G. and Papavasiliou,F.N.\n",
      "  TITLE     Immunoglobulin somatic hypermutation\n",
      "  JOURNAL   Annu Rev Genet 41, 107-120 (2007)\n",
      "   PUBMED   17576170\n",
      "  REMARK    REVIEW ON SOMATIC HYPERMUTATION.\n",
      "REFERENCE   4  (residues 1 to 449)\n",
      "  AUTHORS   Thaysen-Andersen,M., Mysling,S. and Hojrup,P.\n",
      "  TITLE     Site-specific glycoprofiling of N-linked glycopeptides using\n",
      "            MALDI-TOF MS: strong correlation between signal strength and\n",
      "            glycoform quantities\n",
      "  JOURNAL   Anal Chem 81 (10), 3933-3943 (2009)\n",
      "   PUBMED   19358553\n",
      "  REMARK    GLYCOSYLATION AT ASN-299.\n",
      "REFERENCE   5  (residues 1 to 449)\n",
      "  AUTHORS   Schroeder,H.W. Jr. and Cavacini,L.\n",
      "  TITLE     Structure and function of immunoglobulins\n",
      "  JOURNAL   J Allergy Clin Immunol 125 (2 Suppl 2), S41-S52 (2010)\n",
      "   PUBMED   20176268\n",
      "  REMARK    REVIEW ON IMMUNOGLOBULINS.\n",
      "REFERENCE   6  (residues 1 to 449)\n",
      "  AUTHORS   McHeyzer-Williams,M., Okitsu,S., Wang,N. and McHeyzer-Williams,L.\n",
      "  TITLE     Molecular programming of B cell memory\n",
      "  JOURNAL   Nat Rev Immunol 12 (1), 24-34 (2011)\n",
      "   PUBMED   22158414\n",
      "  REMARK    REVIEW ON FUNCTION.\n",
      "            Publication Status: Online-Only\n",
      "COMMENT     On Jul 20, 2018 this sequence version replaced P0DOX5.1.\n",
      "            [FUNCTION] Immunoglobulins, also known as antibodies, are\n",
      "            membrane-bound or secreted glycoproteins produced by B lymphocytes.\n",
      "            In the recognition phase of humoral immunity, the membrane-bound\n",
      "            immunoglobulins serve as receptors which, upon binding of a\n",
      "            specific antigen, trigger the clonal expansion and differentiation\n",
      "            of B lymphocytes into immunoglobulins-secreting plasma cells.\n",
      "            Secreted immunoglobulins mediate the effector phase of humoral\n",
      "            immunity, which results in the elimination of bound antigens\n",
      "            (PubMed:20176268, PubMed:22158414). The antigen binding site is\n",
      "            formed by the variable domain of one heavy chain, together with\n",
      "            that of its associated light chain. Thus, each immunoglobulin has\n",
      "            two antigen binding sites with remarkable affinity for a particular\n",
      "            antigen. The variable domains are assembled by a process called\n",
      "            V-(D)-J rearrangement and can then be subjected to somatic\n",
      "            hypermutations which, after exposure to antigen and selection,\n",
      "            allow affinity maturation for a particular antigen\n",
      "            (PubMed:17576170, PubMed:20176268). {ECO:0000303|PubMed:17576170,\n",
      "            ECO:0000303|PubMed:20176268, ECO:0000303|PubMed:22158414}.\n",
      "            [SUBUNIT] Immunoglobulins are composed of two identical heavy\n",
      "            chains and two identical light chains; disulfide-linked.\n",
      "            {ECO:0000303|PubMed:20176268}.\n",
      "            [SUBCELLULAR LOCATION] Secreted {ECO:0000303|PubMed:20176268,\n",
      "            ECO:0000303|PubMed:22158414}. Cell membrane\n",
      "            {ECO:0000303|PubMed:20176268, ECO:0000303|PubMed:22158414}.\n",
      "            [CAUTION] This sequence is an example of a full-length\n",
      "            immunoglobulin gamma-1 heavy chain. {ECO:0000305}.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..449\n",
      "                     /organism=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:9606\"\n",
      "     Protein         1..449\n",
      "                     /product=\"Immunoglobulin gamma-1 heavy chain\"\n",
      "                     /note=\"Immunoglobulin gamma-1 heavy chain NIE\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..449\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /note=\"Immunoglobulin gamma-1 heavy chain.\n",
      "                     /id=PRO_0000439287.\"\n",
      "     Region          1..119\n",
      "                     /region_name=\"Region of interest in the sequence\"\n",
      "                     /note=\"Variable (V) domain, involved in antigen\n",
      "                     recognition. /evidence=ECO:0000305.\"\n",
      "     Region          1..96\n",
      "                     /region_name=\"Domain\"\n",
      "                     /note=\"Ig-like 1.\n",
      "                     /evidence=ECO:0000255|PROSITE-ProRule:PRU00114.\"\n",
      "     Site            1\n",
      "                     /site_type=\"pyrrolidone-carboxylic-acid\"\n",
      "                     /note=\"Pyrrolidone carboxylic acid.\n",
      "                     /evidence=ECO:0000269|PubMed:826475.\"\n",
      "     Region          3..118\n",
      "                     /region_name=\"IgV_H\"\n",
      "                     /note=\"Immunoglobulin (Ig) heavy chain (H), variable (V)\n",
      "                     domain; cd04981\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          3..24\n",
      "                     /region_name=\"FR1\"\n",
      "                     /note=\"FR1 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          3..6\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          3..6\n",
      "                     /region_name=\"Ig strand A\"\n",
      "                     /note=\"Ig strand A [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Site            order(9,11,114,116,118)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"intrachain IgC domain interface [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          10..13\n",
      "                     /region_name=\"Ig strand A'\"\n",
      "                     /note=\"Ig strand A' [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          10..12\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          16..24\n",
      "                     /region_name=\"Ig strand B\"\n",
      "                     /note=\"Ig strand B [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          18..27\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Bond            bond(22,96)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Region          25..32\n",
      "                     /region_name=\"CDR1\"\n",
      "                     /note=\"CDR1 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          29..31\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Site            order(33,50,99)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"antigen binding site [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          33..44\n",
      "                     /region_name=\"FR2\"\n",
      "                     /note=\"FR2 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          33..40\n",
      "                     /region_name=\"Ig strand C\"\n",
      "                     /note=\"Ig strand C [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          34..39\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Site            order(39,43,47,95,109..110)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"heterodimer interface [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          41..43\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6V8Z.\"\n",
      "     Region          45..61\n",
      "                     /region_name=\"CDR2\"\n",
      "                     /note=\"CDR2 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          45..52\n",
      "                     /region_name=\"Ig strand C'\"\n",
      "                     /note=\"Ig strand C' [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          45..51\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          53..55\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6N35.\"\n",
      "     Region          56..60\n",
      "                     /region_name=\"Ig strand C'\"\n",
      "                     /note=\"Ig strand C' [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          58..60\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          62..64\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          65..67\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          67..98\n",
      "                     /region_name=\"FR3\"\n",
      "                     /note=\"FR3 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          67..73\n",
      "                     /region_name=\"Ig strand D\"\n",
      "                     /note=\"Ig strand D [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          68..73\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Site            73\n",
      "                     /site_type=\"glycosylation\"\n",
      "                     /note=\"N-linked (GlcNAc...) asparagine.\n",
      "                     /evidence=ECO:0000255.\"\n",
      "     Region          74..76\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          78..84\n",
      "                     /region_name=\"Ig strand E\"\n",
      "                     /note=\"Ig strand E [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          78..83\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          88..90\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          92..101\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          92..99\n",
      "                     /region_name=\"Ig strand F\"\n",
      "                     /note=\"Ig strand F [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          99..107\n",
      "                     /region_name=\"CDR3\"\n",
      "                     /note=\"CDR3 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          103..105\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:7CZV.\"\n",
      "     Region          106..109\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          108..118\n",
      "                     /region_name=\"FR4\"\n",
      "                     /note=\"FR4 [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          108..118\n",
      "                     /region_name=\"Ig strand G\"\n",
      "                     /note=\"Ig strand G [structural motif]\"\n",
      "                     /db_xref=\"CDD:409370\"\n",
      "     Region          113..117\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          120..449\n",
      "                     /region_name=\"Region of interest in the sequence\"\n",
      "                     /note=\"Constant (C) domain. /evidence=ECO:0000305.\"\n",
      "     Region          124..217\n",
      "                     /region_name=\"IgC1_CH1_IgEG\"\n",
      "                     /note=\"CH1 domain (first constant Ig domain of the heavy\n",
      "                     chain) in immunoglobulin heavy epsilon and gamma chain;\n",
      "                     member of the C1-set of Ig superfamily (IgSF) domains;\n",
      "                     cd21817\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          124..131\n",
      "                     /region_name=\"Ig strand A\"\n",
      "                     /note=\"Ig strand A [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          125..218\n",
      "                     /region_name=\"Domain\"\n",
      "                     /note=\"Ig-like 2.\n",
      "                     /evidence=ECO:0000255|PROSITE-ProRule:PRU00114.\"\n",
      "     Region          126..130\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Site            order(128..130,143,147,149,170,172,175..176,183..185,187,\n",
      "                     215)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"heterodimer interface [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          134..136\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6UOE.\"\n",
      "     Region          141..154\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          142..149\n",
      "                     /region_name=\"Ig strand B\"\n",
      "                     /note=\"Ig strand B [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Bond            bond(146,202)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Region          155..161\n",
      "                     /region_name=\"Ig strand C\"\n",
      "                     /note=\"Ig strand C [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          157..160\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          161..163\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          163..164\n",
      "                     /region_name=\"Ig strand C'\"\n",
      "                     /note=\"Ig strand C' [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          169..171\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          170..176\n",
      "                     /region_name=\"Ig strand D\"\n",
      "                     /note=\"Ig strand D [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          175..178\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6V8Z.\"\n",
      "     Region          182..191\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          182..190\n",
      "                     /region_name=\"Ig strand E\"\n",
      "                     /note=\"Ig strand E [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          192..194\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          195..197\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          199..206\n",
      "                     /region_name=\"Ig strand F\"\n",
      "                     /note=\"Ig strand F [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Region          201..206\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          207..209\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          211..216\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6ARP.\"\n",
      "     Region          211..216\n",
      "                     /region_name=\"Ig strand G\"\n",
      "                     /note=\"Ig strand G [structural motif]\"\n",
      "                     /db_xref=\"CDD:409622\"\n",
      "     Bond            bond(222)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"Interchain (with light chain).\n",
      "                     /evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Bond            bond(228)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"Interchain (with heavy chain).\n",
      "                     /evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Bond            bond(231)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"Interchain (with heavy chain).\n",
      "                     /evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Region          240..339\n",
      "                     /region_name=\"IgC1_CH3_IgAEM_CH2_IgG\"\n",
      "                     /note=\"CH3 domain (third constant Ig domain of heavy\n",
      "                     chains) in immunoglobulin heavy alpha, epsilon, and mu\n",
      "                     chains, and CH2 domain (second constant Ig domain of the\n",
      "                     gheavy chain) in immunoglobulin heavy gamma chain; member\n",
      "                     of the C1-set of Ig superfamily (IgSF); cd07696\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          240..339\n",
      "                     /region_name=\"Domain\"\n",
      "                     /note=\"Ig-like 3.\n",
      "                     /evidence=ECO:0000255|PROSITE-ProRule:PRU00114.\"\n",
      "     Region          241..247\n",
      "                     /region_name=\"Ig strand A\"\n",
      "                     /note=\"Ig strand A [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          241..245\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Site            order(243..246,260,262,264,266,302..305,307,309..310,\n",
      "                     337..339)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"heterodimer interface [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Site            order(246,248..250)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"intrachain IgC domain interface [polypeptide\n",
      "                     binding]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          249..253\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:5W5L.\"\n",
      "     Region          255..257\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:5O4E.\"\n",
      "     Region          259..266\n",
      "                     /region_name=\"Ig strand B\"\n",
      "                     /note=\"Ig strand B [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          260..268\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Bond            bond(263,323)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Region          270..272\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          274..280\n",
      "                     /region_name=\"Ig strand C\"\n",
      "                     /note=\"Ig strand C [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          276..281\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          283..285\n",
      "                     /region_name=\"Ig strand C'\"\n",
      "                     /note=\"Ig strand C' [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          284..286\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          290..297\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          290..295\n",
      "                     /region_name=\"Ig strand D\"\n",
      "                     /note=\"Ig strand D [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          298..300\n",
      "                     /region_name=\"Hydrogen bonded turn\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Site            299\n",
      "                     /site_type=\"glycosylation\"\n",
      "                     /note=\"N-linked (GlcNAc...) (complex) asparagine.\n",
      "                     /evidence=ECO:0000269|PubMed:19358553.\"\n",
      "     Region          301..309\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          302..310\n",
      "                     /region_name=\"Ig strand E\"\n",
      "                     /note=\"Ig strand E [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          312..316\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          320..327\n",
      "                     /region_name=\"Ig strand F\"\n",
      "                     /note=\"Ig strand F [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          321..326\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          328..332\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          334..338\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          335..338\n",
      "                     /region_name=\"Ig strand G\"\n",
      "                     /note=\"Ig strand G [structural motif]\"\n",
      "                     /db_xref=\"CDD:409493\"\n",
      "     Region          348..449\n",
      "                     /region_name=\"IgC1_CH3_IgAGD_CH4_IgAEM\"\n",
      "                     /note=\"CH3 domain (third constant Ig domain of the heavy\n",
      "                     chain) in immunoglobulin heavy alpha, gamma, and delta\n",
      "                     chains, and CH4 domain (fourth constant Ig domain of the\n",
      "                     heavy chain) in immunoglobulin heavy alpha, epsilon, and\n",
      "                     mu chains; member of the C1-set of I; cd05768\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          348..444\n",
      "                     /region_name=\"Domain\"\n",
      "                     /note=\"Ig-like 4.\n",
      "                     /evidence=ECO:0000255|PROSITE-ProRule:PRU00114.\"\n",
      "     Region          348..354\n",
      "                     /region_name=\"Ig strand A\"\n",
      "                     /note=\"Ig strand A [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          349..353\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Site            order(353,368,372,409,413)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"dimer interface [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          359..361\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          362..377\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          365..373\n",
      "                     /region_name=\"Ig strand B\"\n",
      "                     /note=\"Ig strand B [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Bond            bond(369,427)\n",
      "                     /bond_type=\"disulfide\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:1002129.\"\n",
      "     Region          378..384\n",
      "                     /region_name=\"Ig strand C\"\n",
      "                     /note=\"Ig strand C [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          380..385\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Site            order(384,428,434..435,437..438,440)\n",
      "                     /site_type=\"other\"\n",
      "                     /note=\"receptor binding site [polypeptide binding]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          388..390\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6UGW.\"\n",
      "     Region          391..395\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          393..399\n",
      "                     /region_name=\"Ig strand D\"\n",
      "                     /note=\"Ig strand D [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          402..404\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:8A49.\"\n",
      "     Region          406..415\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          406..414\n",
      "                     /region_name=\"Ig strand E\"\n",
      "                     /note=\"Ig strand E [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          416..420\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          424..431\n",
      "                     /region_name=\"Ig strand F\"\n",
      "                     /note=\"Ig strand F [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          425..430\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          435..437\n",
      "                     /region_name=\"Helical region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "     Region          437..444\n",
      "                     /region_name=\"Ig strand G\"\n",
      "                     /note=\"Ig strand G [structural motif]\"\n",
      "                     /db_xref=\"CDD:409425\"\n",
      "     Region          438..443\n",
      "                     /region_name=\"Beta-strand region\"\n",
      "                     /note=\"/evidence=ECO:0007829|PDB:6G1E.\"\n",
      "ORIGIN      \n",
      "        1 qvqlvqsggg vvqpgrslrl scaasgftfs rytihwvrqa pgkglewvav msyngnnkhy\n",
      "       61 adsvngrfti srndskntly lnmnslrped tavyycarir dtamffahwg qgtlvtvssa\n",
      "      121 stkgpsvfpl apsskstsgg taalgclvkd yfpepvtvsw nsgaltsgvh tfpavlqssg\n",
      "      181 lyslssvvtv pssslgtqty icnvnhkpsn tkvdkkvepk scdkthtcpp cpapellggp\n",
      "      241 svflfppkpk dtlmisrtpe vtcvvvdvsh edpevkfnwy vdgvevhnak tkpreeqyns\n",
      "      301 tyrvvsvltv lhqdwlngke ykckvsnkal papiektisk akgqprepqv ytlppsrdel\n",
      "      361 tknqvsltcl vkgfypsdia vewesngqpe nnykttppvl dsdgsfflys kltvdksrwq\n",
      "      421 qgnvfscsvm healhnhytq kslslspgk\n",
      "//\n",
      "\n",
      "\n",
      "IEP of peptide QVQLVQSGGGVVQPGRSLRLSCAASGFTFSRYTIHWVRQAPGKGLEWVAVMSYNGNNKHYADSVNGRFTISRNDSKNTLYLNMNSLRPEDTAVYYCARIRDTAMFFAHWGQGTLVTVSSASTKGPSVFPLAPSSKSTSGGTAALGCLVKDYFPEPVTVSWNSGALTSGVHTFPAVLQSSGLYSLSSVVTVPSSSLGTQTYICNVNHKPSNTKVDKKVEPKSCDKTHTCPPCPAPELLGGPSVFLFPPKPKDTLMISRTPEVTCVVVDVSHEDPEVKFNWYVDGVEVHNAKTKPREEQYNSTYRVVSVLTVLHQDWLNGKEYKCKVSNKALPAPIEKTISKAKGQPREPQVYTLPPSRDELTKNQVSLTCLVKGFYPSDIAVEWESNGQPENNYKTTPPVLDSDGSFFLYSKLTVDKSRWQQGNVFSCSVMHEALHNHYTQKSLSLSPGK is 8.94\n",
      "It's charge at pH 6.8 is 11.43\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"protein\", term=\"P0DOX5\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq\n",
    "print(record)\n",
    "protein = IP(protein_sequence)\n",
    "\n",
    "\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 6.8 is {:.2f}\".format(protein.charge_at_pH(6.8)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       CASA1_BOVIN              214 aa            linear   MAM 02-OCT-2024\n",
      "DEFINITION  RecName: Full=Alpha-S1-casein; AltName: Allergen=Bos d 8; Contains:\n",
      "            RecName: Full=Antioxidant peptide; Flags: Precursor.\n",
      "ACCESSION   P02662\n",
      "VERSION     P02662.2\n",
      "DBSOURCE    UniProtKB: locus CASA1_BOVIN, accession P02662;\n",
      "            class: standard.\n",
      "            extra\n",
      "            accessions:A5YK80,A8WCP1,A8WCP2,A8WCP3,A8WCP4,A8WCP5,A8WCP6,A8WCP7,\n",
      "            A8WCP8,A8WCP9,A8WCQ0,A8WCQ1,A8WCQ2,A8WCQ3,A8WCQ4,A8WCQ5,A8WCQ6,\n",
      "            A8WCQ7,A8WCQ8,A8WCQ9,A8WCR0,A8WCR1,A8WCR2,A8WCR3,A8WCR4,A8WCR5,\n",
      "            A8WCR6,A8WCR7,A8WCR8,A8WCR9,A8WCS0,A8WCS1,Q28048,Q28069,Q32LE8,\n",
      "            Q7M2U6,Q9TRH5\n",
      "            created: Jul 21, 1986.\n",
      "            sequence updated: Nov 1, 1990.\n",
      "            annotation updated: Oct 2, 2024.\n",
      "            xrefs: X00564.1, CAB57792.1, M33123.1, AAA30428.1, M38641.1,\n",
      "            AAA30429.1, X59856.2, CAA42516.1, EU221551.1, ABW98936.1,\n",
      "            EU221552.1, ABW98937.1, EU221553.1, ABW98938.1, EU221554.1,\n",
      "            ABW98939.1, EU221555.1, ABW98940.1, EU221556.1, ABW98941.1,\n",
      "            EU221557.1, ABW98942.1, EU221558.1, ABW98943.1, EU221559.1,\n",
      "            ABW98944.1, EU221560.1, ABW98945.1, EU221561.1, ABW98946.1,\n",
      "            EU221562.1, ABW98947.1, EU221563.1, ABW98948.1, EU221564.1,\n",
      "            ABW98949.1, EU221565.1, ABW98950.1, EU221566.1, ABW98951.1,\n",
      "            EU221567.1, ABW98952.1, EU221568.1, ABW98953.1, EU221569.1,\n",
      "            ABW98954.1, EU221570.1, ABW98955.1, EU221571.1, ABW98956.1,\n",
      "            EU221572.1, ABW98957.1, EU221573.1, ABW98958.1, EU221574.1,\n",
      "            ABW98959.1, EU221575.1, ABW98960.1, EU221576.1, ABW98961.1,\n",
      "            EU221577.1, ABW98962.1, EU221578.1, ABW98963.1, EU221579.1,\n",
      "            ABW98964.1, EU221580.1, ABW98965.1, EU221581.1, ABW98966.1,\n",
      "            BC109618.1, AAI09619.1, K01084.1, AAA30478.1, EF538766.1,\n",
      "            ABQ88318.1, M38658.1, AAA62707.1, AH007360.2, AAD14099.1, D00412.1,\n",
      "            BAA00313.1, KABOSB, S72220, NP_851372.1\n",
      "            xrefs (non-sequence databases): AlphaFoldDB:P02662, EMDB:EMD-3776,\n",
      "            EMDB:EMD-4622, EMDB:EMD-4623, EMDB:EMD-4625, BioGRID:159482,\n",
      "            DIP:DIP-29917N, IntAct:P02662, MINT:P02662,\n",
      "            STRING:9913.ENSBTAP00000010119, Allergome:10197, Allergome:167,\n",
      "            Allergome:2734, CarbonylDB:P02662, iPTMnet:P02662,\n",
      "            PaxDb:9913-ENSBTAP00000010119, PeptideAtlas:P02662,\n",
      "            Ensembl:ENSBTAT00000010119.4, Ensembl:ENSBTAP00000010119.2,\n",
      "            Ensembl:ENSBTAG00000007695.6, GeneID:282208, KEGG:bta:282208,\n",
      "            CTD:1446, VEuPathDB:HostDB:ENSBTAG00000007695, VGNC:27763,\n",
      "            eggNOG:ENOG502TEWT, GeneTree:ENSGT00390000017378,\n",
      "            HOGENOM:CLU_126070_0_0_1, InParanoid:P02662, OMA:AYLPAPW,\n",
      "            TreeFam:TF340763, Reactome:R-BTA-5223345, PRO:PR:P02662,\n",
      "            Proteomes:UP000009136, Bgee:ENSBTAG00000007695, GO:0005615,\n",
      "            GO:0005794, GO:0005796, GO:0001540, GO:0016209, GO:1902904,\n",
      "            GO:0050821, GO:1903496, GO:1903494, GO:0032355, GO:0060416,\n",
      "            GO:0032570, InterPro:IPR026999, InterPro:IPR001588,\n",
      "            InterPro:IPR031305, PANTHER:PTHR10240, PANTHER:PTHR10240:SF0,\n",
      "            Pfam:PF00363, PROSITE:PS00306\n",
      "KEYWORDS    Allergen; Antioxidant; Direct protein sequencing; Milk protein;\n",
      "            Phosphoprotein; Reference proteome; Repeat; Secreted; Signal.\n",
      "SOURCE      Bos taurus (domestic cattle)\n",
      "  ORGANISM  Bos taurus\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Laurasiatheria; Artiodactyla; Ruminantia;\n",
      "            Pecora; Bovidae; Bovinae; Bos.\n",
      "REFERENCE   1  (residues 1 to 214)\n",
      "  AUTHORS   Stewart,A.F., Willis,I.M. and Mackinlay,A.G.\n",
      "  TITLE     Nucleotide sequences of bovine alpha S1- and kappa-casein cDNAs\n",
      "  JOURNAL   Nucleic Acids Res 12 (9), 3895-3907 (1984)\n",
      "   PUBMED   6328443\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA].\n",
      "REFERENCE   2  (residues 1 to 214)\n",
      "  AUTHORS   Nagao,M., Maki,M., Sasaki,R. and Chiba,R.\n",
      "  TITLE     Isolation and sequence analysis of bovine alpha-S1-casein cDNA\n",
      "            clone\n",
      "  JOURNAL   Agric. Biol. Chem. 48, 1663-1667 (1984)\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA].\n",
      "REFERENCE   3  (residues 1 to 214)\n",
      "  AUTHORS   Gorodetskii,S.I., Zakhar'ev,V.M., Kiarshulite,D.R.,\n",
      "            Kapelinskaia,T.V. and Skriabin,K.G.\n",
      "  TITLE     [cDNA of cattle alpha S1-casein: cloning and nucleotide sequence]\n",
      "  JOURNAL   Biokhimiia 51 (10), 1641-1648 (1986)\n",
      "   PUBMED   3022833\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA].;\n",
      "            TISSUE=Mammary gland\n",
      "REFERENCE   4  (residues 1 to 214)\n",
      "  AUTHORS   Koczan,D., Hobom,G. and Seyfert,H.M.\n",
      "  TITLE     Genomic organization of the bovine alpha-S1 casein gene\n",
      "  JOURNAL   Nucleic Acids Res 19 (20), 5591-5596 (1991)\n",
      "   PUBMED   1658736\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA].\n",
      "REFERENCE   5  (residues 1 to 214)\n",
      "  AUTHORS   Schulmeister,U., Hochwallner,H., Swoboda,I., Focke-Tejkl,M.,\n",
      "            Geller,B., Nystrand,M., Harlin,A., Thalhamer,J., Scheiblhofer,S.,\n",
      "            Keller,W., Niggemann,B., Quirce,S., Ebner,C., Mari,A., Pauli,G.,\n",
      "            Herz,U., Valenta,R. and Spitzauer,S.\n",
      "  TITLE     Cloning, expression, and mapping of allergenic determinants of\n",
      "            alphaS1-casein, a major cow's milk allergen\n",
      "  JOURNAL   J Immunol 182 (11), 7019-7029 (2009)\n",
      "   PUBMED   19454699\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA], AND ALLERGEN.\n",
      "REFERENCE   6  (residues 1 to 214)\n",
      "  CONSRTM   NIH - Mammalian Gene Collection (MGC) project\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-NOV-2005) to the EMBL/GenBank/DDBJ databases\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA].;\n",
      "            STRAIN=Crossbred X Angus; TISSUE=Liver\n",
      "REFERENCE   7  (residues 1 to 214)\n",
      "  AUTHORS   Mercier,J.C., Grosclaude,F. and Ribadeau-Dumas,B.\n",
      "  TITLE     [Primary structure of bovine s1 casein. Complete sequence]\n",
      "  JOURNAL   Eur J Biochem 23 (1), 41-51 (1971)\n",
      "   PUBMED   4331376\n",
      "  REMARK    PROTEIN SEQUENCE OF 16-214 (VARIANT B).\n",
      "REFERENCE   8  (residues 1 to 214)\n",
      "  AUTHORS   Grosclaude,F., Mahe,M.F. and Ribadeau-Dumas,B.\n",
      "  TITLE     [Primary structure of alpha casein and of bovine beta casein.\n",
      "            Correction]\n",
      "  JOURNAL   Eur J Biochem 40 (1), 323-324 (1973)\n",
      "   PUBMED   4797901\n",
      "  REMARK    SEQUENCE REVISION TO 74 AND 92-93 (VARIANTS A; B; C AND D).\n",
      "REFERENCE   9  (residues 1 to 214)\n",
      "  AUTHORS   Grosclaude,F., Mahe,M.F., Mercier,J.C. and Ribadeau-Dumas,B.\n",
      "  TITLE     [Characterization of genetic variants of a S1 and bovine caseins]\n",
      "  JOURNAL   Eur J Biochem 26 (3), 328-337 (1972)\n",
      "   PUBMED   5064450\n",
      "  REMARK    PROTEIN SEQUENCE OF 16-214 (VARIANT D).\n",
      "REFERENCE   10 (residues 1 to 214)\n",
      "  AUTHORS   Grosclaude,F., Mahe,M.F., Mercier,J.C. and Ribadeau-Dumas,B.\n",
      "  TITLE     [Not Available]\n",
      "  JOURNAL   FEBS Lett 11 (2), 109-112 (1970)\n",
      "   PUBMED   11945462\n",
      "  REMARK    PROTEIN SEQUENCE OF 23-49 (VARIANT A).\n",
      "REFERENCE   11 (residues 1 to 214)\n",
      "  AUTHORS   Neuteboom,B., Giuffrida,M.G. and Conti,A.\n",
      "  TITLE     Isolation of a new ligand-carrying casein fragment from bovine\n",
      "            mammary gland microsomes\n",
      "  JOURNAL   FEBS Lett 305 (3), 189-191 (1992)\n",
      "   PUBMED   1299613\n",
      "  REMARK    PROTEIN SEQUENCE OF 39-55.;\n",
      "            TISSUE=Mammary gland\n",
      "REFERENCE   12 (residues 1 to 214)\n",
      "  AUTHORS   Ramalho-Santos,M., Verissimo,P., Faro,C. and Pires,E.\n",
      "  TITLE     Action on bovine alpha s1-casein of cardosins A and B, aspartic\n",
      "            proteinases from the flowers of the cardoon Cynara cardunculus L\n",
      "  JOURNAL   Biochim Biophys Acta 1297 (1), 83-89 (1996)\n",
      "   PUBMED   8841384\n",
      "  REMARK    PROTEIN SEQUENCE OF 39-47 AND 166-188.\n",
      "REFERENCE   13 (residues 1 to 214)\n",
      "  AUTHORS   Willis,I.M., Stewart,A.F., Caputo,A., Thompson,A.R. and\n",
      "            Mackinlay,A.G.\n",
      "  TITLE     Construction and identification by partial nucleotide sequence\n",
      "            analysis of bovine casein and beta-lactoglobulin cDNA clones\n",
      "  JOURNAL   DNA 1 (4), 375-386 (1982)\n",
      "   PUBMED   6897774\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA] OF 55-130.\n",
      "REFERENCE   14 (residues 1 to 214)\n",
      "  AUTHORS   Shahla,M.N., Farooq,M.S., Naeem,M.K. and Riazuddin,S.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-APR-2007) to the EMBL/GenBank/DDBJ databases\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA] OF 85-110.\n",
      "REFERENCE   15 (residues 1 to 214)\n",
      "  AUTHORS   Gupta,A., Mann,B., Kumar,R. and Sangwan,R.B.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-MAR-2008) to UniProtKB\n",
      "  REMARK    PROTEIN SEQUENCE OF 95-105, AND FUNCTION.\n",
      "REFERENCE   16 (residues 1 to 214)\n",
      "  AUTHORS   Kiarshulite,D.R., Zakhar'ev,V.M. and Gorodetskii,S.I.\n",
      "  TITLE     [Nucleotide sequence of the 3'-nontranslated region of the mRNA of\n",
      "            alpha S1-casein in cows]\n",
      "  JOURNAL   Dokl Akad Nauk SSSR 280 (6), 1433-1437 (1985)\n",
      "   PUBMED   3838718\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA] OF 122-214.\n",
      "REFERENCE   17 (residues 1 to 214)\n",
      "  AUTHORS   Chen,R., Wang,B., Zhang,Y., Liu,W., Zhang,J. and Lao,W.\n",
      "  TITLE     Cloning, mapping, and sequencing of 3' and its flanking region of\n",
      "            bovine alpha-s1 casein gene\n",
      "  JOURNAL   Chin J Biotechnol 8 (4), 235-245 (1992)\n",
      "   PUBMED   1343827\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA] OF 164-214.\n",
      "REFERENCE   18 (residues 1 to 214)\n",
      "  AUTHORS   Maki,M., Nagao,M., Hirose,M. and Chiba,H.\n",
      "  TITLE     Cloning of cDNA sequence coding for bovine alpha-s1-casein\n",
      "  JOURNAL   Agric. Biol. Chem. 47, 441-444 (1983)\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA] OF 185-202.;\n",
      "            TISSUE=Mammary gland\n",
      "REFERENCE   19 (residues 1 to 214)\n",
      "  AUTHORS   Grosclaude,F., Mercier,J.-C. and Ribadeau-Dumas,B.\n",
      "  TITLE     On the localization in the C-terminal sequence of bovine casein\n",
      "            alpha-s1 of a Glu/Gly substitution that differentiates the genetic\n",
      "            variants B and C\n",
      "  JOURNAL   C. R. Hebd. Seances Acad. Sci., D, Sci. Nat. 268, 3133-3136 (1970)\n",
      "  REMARK    PROTEIN SEQUENCE OF 205-214 (VARIANT C).\n",
      "REFERENCE   20 (residues 1 to 214)\n",
      "  AUTHORS   Chatchatee,P., Jarvinen,K.M., Bardina,L., Beyer,K. and Sampson,H.A.\n",
      "  TITLE     Identification of IgE- and IgG-binding epitopes on\n",
      "            alpha(s1)-casein: differences in patients with persistent and\n",
      "            transient cow's milk allergy\n",
      "  JOURNAL   J Allergy Clin Immunol 107 (2), 379-383 (2001)\n",
      "   PUBMED   11174208\n",
      "  REMARK    ALLERGEN.\n",
      "REFERENCE   21 (residues 1 to 214)\n",
      "  AUTHORS   Larsen,M.R., Thingholm,T.E., Jensen,O.N., Roepstorff,P. and\n",
      "            Jorgensen,T.J.\n",
      "  TITLE     Highly selective enrichment of phosphorylated peptides from peptide\n",
      "            mixtures using titanium dioxide microcolumns\n",
      "  JOURNAL   Mol Cell Proteomics 4 (7), 873-886 (2005)\n",
      "   PUBMED   15858219\n",
      "  REMARK    PHOSPHORYLATION AT SER-56; SER-61; SER-63; SER-79; SER-81; SER-82;\n",
      "            SER-83; SER-90 AND SER-130, AND IDENTIFICATION BY MASS\n",
      "            SPECTROMETRY.\n",
      "REFERENCE   22 (residues 1 to 214)\n",
      "  AUTHORS   Bai,F., Liu,S. and Witzmann,F.A.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-SEP-2005) to UniProtKB\n",
      "  REMARK    PHOSPHORYLATION AT SER-56.\n",
      "COMMENT     On or before May 6, 2008 this sequence version replaced\n",
      "            gi:75039462, gi:75045423, gi:75051105, gi:1070620.\n",
      "            [FUNCTION] Important role in the capacity of milk to transport\n",
      "            calcium phosphate. {ECO:0000269|Ref.15}.\n",
      "            [FUNCTION] Antioxidant peptide has 2,2-diphenyl-1-picrylhydrazyl\n",
      "            (DPPH) radical scavenging activity. {ECO:0000269|Ref.15}.\n",
      "            [SUBCELLULAR LOCATION] Secreted.\n",
      "            [TISSUE SPECIFICITY] Mammary gland specific. Secreted in milk.\n",
      "            [ALLERGEN] Causes an allergic reaction in human. Has six major and\n",
      "            3 minor IgE-binding regions and 5 major and 1 minor IgG-binding\n",
      "            regions. Is a cause of cow's milk allergy (CMA).\n",
      "            {ECO:0000269|PubMed:11174208, ECO:0000269|PubMed:19454699}.\n",
      "            [MISCELLANEOUS] The B variant sequence is shown.\n",
      "            [SIMILARITY] Belongs to the alpha-casein family. {ECO:0000305}.\n",
      "            [WEB RESOURCE] Name=Worthington enzyme manual;\n",
      "            URL='https://www.worthington-biochem.com/CASA/'.\n",
      "            [WEB RESOURCE] Name=Protein Spotlight; Note=Of buttons, digestion\n",
      "            and glue - Issue 16 of November 2001;\n",
      "            URL='https://web.expasy.org/spotlight/back_issues/016'.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..214\n",
      "                     /organism=\"Bos taurus\"\n",
      "                     /db_xref=\"taxon:9913\"\n",
      "     gene            1..214\n",
      "                     /gene=\"CSN1S1\"\n",
      "     Protein         1..214\n",
      "                     /product=\"Alpha-S1-casein\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..15\n",
      "                     /region_name=\"Signal\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:4331376,\n",
      "                     ECO:0000269|PubMed:5064450.\"\n",
      "     Region          11..12\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"AV -> SA (in Ref. 5; ABW98943).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          16..214\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /note=\"Alpha-S1-casein. /id=PRO_0000004446.\"\n",
      "     Region          29..41\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"Missing (in variant A).\"\n",
      "     Region          42\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"P -> L (in Ref. 3; AAA30429).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          44\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"P -> S (in Ref. 5; ABW98945).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Site            56\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219,\n",
      "                     ECO:0000269|Ref.22.\"\n",
      "     Site            61\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Site            63\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Region          68\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"A -> T (in variant D).\"\n",
      "     Site            79\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Site            81\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Site            82\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Site            83\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Region          85..99\n",
      "                     /region_name=\"Repetitive region\"\n",
      "     Site            90\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Region          95..105\n",
      "                     /region_name=\"Processed active peptide\"\n",
      "                     /note=\"Antioxidant peptide. /id=PRO_0000331578.\"\n",
      "     Region          95\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"H -> Q (in Ref. 13; AAA30478).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          99\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"E -> D (in Ref. 14; ABQ88318).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          112..204\n",
      "                     /region_name=\"Casein\"\n",
      "                     /note=\"pfam00363\"\n",
      "                     /db_xref=\"CDD:459782\"\n",
      "     Region          125..140\n",
      "                     /region_name=\"Repetitive region\"\n",
      "     Site            130\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine.\n",
      "                     /evidence=ECO:0000269|PubMed:15858219.\"\n",
      "     Region          143\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"H -> D (in Ref. 3; AAA30429).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          171\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"L -> P (in Ref. 5; ABW98953).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          195..214\n",
      "                     /region_name=\"Region of interest in the sequence\"\n",
      "                     /note=\"Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite.\"\n",
      "     Region          203\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"S -> L (in Ref. 16; AAA62707).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          207\n",
      "                     /region_name=\"Variant\"\n",
      "                     /note=\"E -> G (in variant C).\"\n",
      "     Region          211..212\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"MP -> IS (in Ref. 3; AAA30429).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "ORIGIN      \n",
      "        1 mklliltclv avalarpkhp ikhqglpqev lnenllrffv apfpevfgke kvnelskdig\n",
      "       61 sestedqame dikqmeaesi ssseeivpns veqkhiqked vpserylgyl eqllrlkkyk\n",
      "      121 vpqleivpns aeerlhsmke gihaqqkepm igvnqelayf ypelfrqfyq ldaypsgawy\n",
      "      181 yvplgtqytd apsfsdipnp igsensektt mplw\n",
      "//\n",
      "\n",
      "\n",
      "IEP of peptide MKLLILTCLVAVALARPKHPIKHQGLPQEVLNENLLRFFVAPFPEVFGKEKVNELSKDIGSESTEDQAMEDIKQMEAESISSSEEIVPNSVEQKHIQKEDVPSERYLGYLEQLLRLKKYKVPQLEIVPNSAEERLHSMKEGIHAQQKEPMIGVNQELAYFYPELFRQFYQLDAYPSGAWYYVPLGTQYTDAPSFSDIPNPIGSENSEKTTMPLW is 4.98\n",
      "It's charge at pH 6.8 is -10.63\n"
     ]
    }
   ],
   "source": [
    "handle = Entrez.esearch(db=\"protein\", term=\"P02662\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq\n",
    "protein = IP(protein_sequence)\n",
    "print(record)\n",
    "\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 6.8 is {:.2f}\".format(protein.charge_at_pH(6.8)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LOCUS       CASB_HUMAN               226 aa            linear   PRI 02-OCT-2024\n",
      "DEFINITION  RecName: Full=Beta-casein; Flags: Precursor.\n",
      "ACCESSION   P05814\n",
      "VERSION     P05814.4\n",
      "DBSOURCE    UniProtKB: locus CASB_HUMAN, accession P05814;\n",
      "            class: standard.\n",
      "            extra accessions:Q4VAZ9,Q9UCM5\n",
      "            created: Nov 1, 1988.\n",
      "            sequence updated: Aug 1, 1992.\n",
      "            annotation updated: Oct 2, 2024.\n",
      "            xrefs: X55739.1, CAA39270.1, AF027807.1, AAC82978.1, X17070.1,\n",
      "            CAA34916.1, BC069554.1, AAH69554.1, BC096194.3, AAH96194.1,\n",
      "            BC096195.1, AAH96195.1, BC096196.3, AAH96196.1, BC096197.3,\n",
      "            AAH96197.1, X13766.1, CAA32017.1, KBHU, NP_001289699.1,\n",
      "            NP_001882.1, XP_016863249.1\n",
      "            xrefs (non-sequence databases): CCDS:CCDS3532.1,\n",
      "            AlphaFoldDB:P05814, SMR:P05814, BioGRID:107834, IntAct:P05814,\n",
      "            MINT:P05814, STRING:9606.ENSP00000341030, Allergome:1064,\n",
      "            iPTMnet:P05814, PhosphoSitePlus:P05814, BioMuta:CSN2, DMDM:115661,\n",
      "            MassIVE:P05814, PaxDb:9606-ENSP00000341030, PeptideAtlas:P05814,\n",
      "            ProteomicsDB:51861, Antibodypedia:24273, DNASU:1447,\n",
      "            Ensembl:ENST00000353151.4, Ensembl:ENSP00000341030.3,\n",
      "            Ensembl:ENSG00000135222.7, GeneID:1447, KEGG:hsa:1447,\n",
      "            MANE-Select:ENST00000353151.4, AGR:HGNC:2447, CTD:1447,\n",
      "            DisGeNET:1447, GeneCards:CSN2, HGNC:2447, HPA:ENSG00000135222,\n",
      "            MIM:115460, neXtProt:NX_P05814, OpenTargets:ENSG00000135222,\n",
      "            PharmGKB:PA26950, VEuPathDB:HostDB:ENSG00000135222,\n",
      "            eggNOG:ENOG502RU0R, GeneTree:ENSGT00390000001890,\n",
      "            HOGENOM:CLU_106775_0_0_1, InParanoid:P05814, OMA:EIMEVPK,\n",
      "            OrthoDB:4933674at2759, PhylomeDB:P05814, TreeFam:TF336929,\n",
      "            PathwayCommons:P05814, Reactome:R-HSA-1251985, SignaLink:P05814,\n",
      "            BioGRID-ORCS:1447, GeneWiki:CSN2, GenomeRNAi:1447, Pharos:P05814,\n",
      "            PRO:PR:P05814, Proteomes:UP000005640, RNAct:P05814,\n",
      "            Bgee:ENSG00000135222, ExpressionAtlas:P05814, GO:0005576,\n",
      "            GO:0005615, GO:0005509, GO:0004869, GO:0004857, GO:0006816,\n",
      "            GO:0007595, GO:2000117, InterPro:IPR001588, InterPro:IPR016345,\n",
      "            InterPro:IPR031305, PANTHER:PTHR11500, PANTHER:PTHR11500:SF0,\n",
      "            Pfam:PF00363, PIRSF:PIRSF002372, PROSITE:PS00306\n",
      "KEYWORDS    Direct protein sequencing; Milk protein; Phosphoprotein; Proteomics\n",
      "            identification; Reference proteome; Secreted; Signal.\n",
      "SOURCE      Homo sapiens (human)\n",
      "  ORGANISM  Homo sapiens\n",
      "            Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;\n",
      "            Mammalia; Eutheria; Euarchontoglires; Primates; Haplorrhini;\n",
      "            Catarrhini; Hominidae; Homo.\n",
      "REFERENCE   1  (residues 1 to 226)\n",
      "  AUTHORS   Lonnerdal,B., Bergstrom,S., Andersson,Y., Hjalmarsson,K.,\n",
      "            Sundqvist,A.K. and Hernell,O.\n",
      "  TITLE     Cloning and sequencing of a cDNA encoding human milk beta-casein\n",
      "  JOURNAL   FEBS Lett 269 (1), 153-156 (1990)\n",
      "   PUBMED   2387396\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA].\n",
      "REFERENCE   2  (residues 1 to 226)\n",
      "  AUTHORS   Hansson,L., Edlund,A., Johansson,T., Hernell,O., Stromqvist,M.,\n",
      "            Lindquist,S., Lonnerdal,B. and Bergstrom,S.\n",
      "  TITLE     Structure of the human beta-casein encoding gene\n",
      "  JOURNAL   Gene 139 (2), 193-199 (1994)\n",
      "   PUBMED   8112603\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA].;\n",
      "            TISSUE=Placenta\n",
      "REFERENCE   3  (residues 1 to 226)\n",
      "  AUTHORS   Menon,R.S.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-OCT-1989) to the EMBL/GenBank/DDBJ databases\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA].;\n",
      "            TISSUE=Mammary gland\n",
      "REFERENCE   4  (residues 1 to 226)\n",
      "  AUTHORS   Kwiatkowski,D.J.\n",
      "  TITLE     Direct Submission\n",
      "  JOURNAL   Submitted (??-DEC-1997) to the EMBL/GenBank/DDBJ databases\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [GENOMIC DNA].;\n",
      "            TISSUE=Placenta\n",
      "REFERENCE   5  (residues 1 to 226)\n",
      "  AUTHORS   Gerhard,D.S., Wagner,L., Feingold,E.A., Shenmen,C.M., Grouse,L.H.,\n",
      "            Schuler,G., Klein,S.L., Old,S., Rasooly,R., Good,P., Guyer,M.,\n",
      "            Peck,A.M., Derge,J.G., Lipman,D., Collins,F.S., Jang,W., Sherry,S.,\n",
      "            Feolo,M., Misquitta,L., Lee,E., Rotmistrovsky,K., Greenhut,S.F.,\n",
      "            Schaefer,C.F., Buetow,K., Bonner,T.I., Haussler,D., Kent,J.,\n",
      "            Kiekhaus,M., Furey,T., Brent,M., Prange,C., Schreiber,K.,\n",
      "            Shapiro,N., Bhat,N.K., Hopkins,R.F., Hsie,F., Driscoll,T.,\n",
      "            Soares,M.B., Casavant,T.L., Scheetz,T.E., Brown-stein,M.J.,\n",
      "            Usdin,T.B., Toshiyuki,S., Carninci,P., Piao,Y., Dudekula,D.B.,\n",
      "            Ko,M.S., Kawakami,K., Suzuki,Y., Sugano,S., Gruber,C.E.,\n",
      "            Smith,M.R., Simmons,B., Moore,T., Waterman,R., Johnson,S.L.,\n",
      "            Ruan,Y., Wei,C.L., Mathavan,S., Gunaratne,P.H., Wu,J., Garcia,A.M.,\n",
      "            Hulyk,S.W., Fuh,E., Yuan,Y., Sneed,A., Kowis,C., Hodgson,A.,\n",
      "            Muzny,D.M., McPherson,J., Gibbs,R.A., Fahey,J., Helton,E.,\n",
      "            Ketteman,M., Madan,A., Rodrigues,S., Sanchez,A., Whiting,M.,\n",
      "            Madari,A., Young,A.C., Wetherby,K.D., Granite,S.J., Kwong,P.N.,\n",
      "            Brinkley,C.P., Pearson,R.L., Bouffard,G.G., Blakesly,R.W.,\n",
      "            Green,E.D., Dickson,M.C., Rodriguez,A.C., Grimwood,J., Schmutz,J.,\n",
      "            Myers,R.M., Butterfield,Y.S., Griffith,M., Griffith,O.L.,\n",
      "            Krzywinski,M.I., Liao,N., Morin,R., Palmquist,D., Petrescu,A.S.,\n",
      "            Skalska,U., Smailus,D.E., Stott,J.M., Schnerch,A., Schein,J.E.,\n",
      "            Jones,S.J., Holt,R.A., Baross,A., Marra,M.A., Clifton,S.,\n",
      "            Makowski,K.A., Bosak,S. and Malek,J.\n",
      "  CONSRTM   MGC Project Team\n",
      "  TITLE     The status, quality, and expansion of the NIH full-length cDNA\n",
      "            project: the Mammalian Gene Collection (MGC)\n",
      "  JOURNAL   Genome Res 14 (10B), 2121-2127 (2004)\n",
      "   PUBMED   15489334\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [LARGE SCALE MRNA].\n",
      "            Erratum:[Genome Res. 2006 Jun;16(6):804. Morrin, Ryan [corrected to\n",
      "            Morin, Ryan]]\n",
      "REFERENCE   6  (residues 1 to 226)\n",
      "  AUTHORS   Greenberg,R., Groves,M.L. and Dower,H.J.\n",
      "  TITLE     Human beta-casein. Amino acid sequence and identification of\n",
      "            phosphorylation sites\n",
      "  JOURNAL   J Biol Chem 259 (8), 5132-5138 (1984)\n",
      "   PUBMED   6715339\n",
      "  REMARK    PROTEIN SEQUENCE OF 16-226, AND PHOSPHORYLATION AT THR-18; SER-21;\n",
      "            SER-23; SER-24 AND SER-25.\n",
      "REFERENCE   7  (residues 1 to 226)\n",
      "  AUTHORS   Poth,A.G., Deeth,H.C., Alewood,P.F. and Holland,J.W.\n",
      "  TITLE     Analysis of the human casein phosphoproteome by 2-D electrophoresis\n",
      "            and MALDI-TOF/TOF MS reveals new phosphoforms\n",
      "  JOURNAL   J Proteome Res 7 (11), 5017-5027 (2008)\n",
      "   PUBMED   18847231\n",
      "  REMARK    PROTEIN SEQUENCE OF 18-33, PHOSPHORYLATION AT THR-18; SER-21;\n",
      "            SER-23; SER-24 AND SER-25, AND IDENTIFICATION BY MASS SPECTROMETRY.\n",
      "REFERENCE   8  (residues 1 to 226)\n",
      "  AUTHORS   Menon,R.S. and Ham,R.G.\n",
      "  TITLE     Human beta-casein: partial cDNA sequence and apparent polymorphism\n",
      "  JOURNAL   Nucleic Acids Res 17 (7), 2869 (1989)\n",
      "   PUBMED   2717418\n",
      "  REMARK    NUCLEOTIDE SEQUENCE [MRNA] OF 161-226.;\n",
      "            TISSUE=Mammary gland\n",
      "REFERENCE   9  (residues 1 to 226)\n",
      "  AUTHORS   Azuma,N. and Yamauchi,K.\n",
      "  TITLE     Plasmin cleavage of human beta-casein\n",
      "  JOURNAL   Biosci Biotechnol Biochem 56 (7), 1140-1141 (1992)\n",
      "   PUBMED   1369431\n",
      "  REMARK    PROTEIN SEQUENCE OF 176-215.\n",
      "COMMENT     On or before Feb 13, 2009 this sequence version replaced\n",
      "            gi:74720247, gi:74753680, gi:7428732.\n",
      "            [FUNCTION] Important role in determination of the surface\n",
      "            properties of the casein micelles.\n",
      "            [INTERACTION] P05814; Q14192: FHL2; NbExp=4; IntAct=EBI-1642112,\n",
      "            EBI-701903; P05814; Q9UHD9: UBQLN2; NbExp=6; IntAct=EBI-1642112,\n",
      "            EBI-947187.\n",
      "            [SUBCELLULAR LOCATION] Secreted.\n",
      "            [TISSUE SPECIFICITY] Mammary gland specific. Secreted in milk.\n",
      "            [PTM] Form 1-P is phosphorylated once; half of the molecules are\n",
      "            phosphorylated on Ser-24, half on Ser-25.\n",
      "            {ECO:0000269|PubMed:18847231, ECO:0000269|PubMed:6715339}.\n",
      "            [SIMILARITY] Belongs to the beta-casein family. {ECO:0000305}.\n",
      "            [WEB RESOURCE] Name=Protein Spotlight; Note=Of buttons, digestion\n",
      "            and glue - Issue 16 of November 2001;\n",
      "            URL='https://web.expasy.org/spotlight/back_issues/016'.\n",
      "FEATURES             Location/Qualifiers\n",
      "     source          1..226\n",
      "                     /organism=\"Homo sapiens\"\n",
      "                     /db_xref=\"taxon:9606\"\n",
      "     gene            1..226\n",
      "                     /gene=\"CSN2\"\n",
      "                     /gene_synonym=\"CASB\"\n",
      "     Protein         1..226\n",
      "                     /product=\"Beta-casein\"\n",
      "                     /UniProtKB_evidence=\"Evidence at protein level\"\n",
      "     Region          1..15\n",
      "                     /region_name=\"Signal\"\n",
      "                     /note=\"/evidence=ECO:0000269|PubMed:6715339.\"\n",
      "     Region          16..226\n",
      "                     /region_name=\"Mature chain\"\n",
      "                     /note=\"Beta-casein. /evidence=ECO:0000269|PubMed:6715339.\n",
      "                     /id=PRO_0000004475.\"\n",
      "     Site            18\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphothreonine; in form 5-P.\n",
      "                     /evidence=ECO:0000269|PubMed:18847231,\n",
      "                     ECO:0000269|PubMed:6715339.\"\n",
      "     Site            21\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine; in form 4-P and form 5-P.\n",
      "                     /evidence=ECO:0000269|PubMed:18847231,\n",
      "                     ECO:0000269|PubMed:6715339.\"\n",
      "     Site            23\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine; in form 3-P, form 4-P and form 5-P.\n",
      "                     /evidence=ECO:0000269|PubMed:18847231,\n",
      "                     ECO:0000269|PubMed:6715339.\"\n",
      "     Site            24\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine; in form 1-P, form 2-P, form 3-P,\n",
      "                     form 4-P and form 5-P.\n",
      "                     /evidence=ECO:0000269|PubMed:18847231,\n",
      "                     ECO:0000269|PubMed:6715339.\"\n",
      "     Site            25\n",
      "                     /site_type=\"phosphorylation\"\n",
      "                     /note=\"Phosphoserine; in form 1-P, form 2-P, form 3-P,\n",
      "                     form 4-P and form 5-P.\n",
      "                     /evidence=ECO:0000269|PubMed:18847231,\n",
      "                     ECO:0000269|PubMed:6715339.\"\n",
      "     Region          30\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"T -> P (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          34\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"Missing (in Ref. 1; CAA39270).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          48..50\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"EDE -> TDQ (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          120\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"S -> Q (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          129..>195\n",
      "                     /region_name=\"Casein\"\n",
      "                     /note=\"pfam00363\"\n",
      "                     /db_xref=\"CDD:459782\"\n",
      "     Region          133\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"L -> V (in Ref. 3; CAA34916).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          140\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"H -> Q (in Ref. 3; CAA34916).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          149\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"L -> S (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          173\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"Q -> E (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          182..184\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"QVV -> EVL (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          188\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"Q -> V (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          207\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"T -> P (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "     Region          214..222\n",
      "                     /region_name=\"Conflict\"\n",
      "                     /note=\"TQPLAPVHN -> PEPSTTZABH (in Ref. 6; AA sequence).\n",
      "                     /evidence=ECO:0000305.\"\n",
      "ORIGIN      \n",
      "        1 mkvlilaclv alalaretie slssseesit eykqkvekvk hedqqqgede hqdkiypsfq\n",
      "       61 pqpliypfve pipygflpqn ilplaqpavv lpvpqpeime vpkakdtvyt kgrvmpvlks\n",
      "      121 ptipffdpqi pkltdlenlh lplpllqplm qqvpqpipqt lalppqplws vpqpkvlpip\n",
      "      181 qqvvpypqra vpvqalllnq elllnpthqi ypvtqplapv hnpisv\n",
      "//\n",
      "\n",
      "\n",
      "IEP of peptide MKVLILACLVALALARETIESLSSSEESITEYKQKVEKVKHEDQQQGEDEHQDKIYPSFQPQPLIYPFVEPIPYGFLPQNILPLAQPAVVLPVPQPEIMEVPKAKDTVYTKGRVMPVLKSPTIPFFDPQIPKLTDLENLHLPLPLLQPLMQQVPQPIPQTLALPPQPLWSVPQPKVLPIPQQVVPYPQRAVPVQALLLNQELLLNPTHQIYPVTQPLAPVHNPISV is 5.52\n",
      "It's charge at pH 6.8 is -4.67\n"
     ]
    }
   ],
   "source": [
    "\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"P05814\", retmax=1)\n",
    "record = Entrez.read(handle)\n",
    "handle.close()\n",
    "\n",
    "# Get the first protein ID from the search results\n",
    "protein_id = record[\"IdList\"][0]\n",
    "\n",
    "# Fetch the protein sequence by ID\n",
    "handle = Entrez.efetch(db=\"protein\", id=protein_id, rettype=\"gb\", retmode=\"text\")\n",
    "record = handle.read()\n",
    "handle.close()\n",
    "\n",
    "seq_record = SeqIO.read(StringIO(record), \"genbank\")\n",
    "protein_sequence = seq_record.seq\n",
    "protein = IP(protein_sequence)\n",
    "print(record)\n",
    "\n",
    "print(\"IEP of peptide {} is {:.2f}\".format(protein.sequence, protein.pi()))\n",
    "\n",
    "print(\"It's charge at pH 6.8 is {:.2f}\".format(protein.charge_at_pH(6.8)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
